2017-09-19 15:37:20,424 ERROR 16594 [-/::1/-/11ms GET /] nodejs.AssertionError [ERR_ASSERTION]: Can't find dist/index.html from 
    at Map.resolve (/Users/caigehui/Documents/ezz-example/node_modules/egg-view/lib/view_manager.js:74:5)
    at resolve.next (<anonymous>)
    at onFulfilled (/Users/caigehui/Documents/ezz-example/node_modules/co/index.js:65:19)
    at <anonymous>
    at process._tickCallback (internal/process/next_tick.js:188:7)
generatedMessage: false
name: 'AssertionError [ERR_ASSERTION]'
code: 'ERR_ASSERTION'
actual: undefined
expected: true
operator: '=='
pid: 16594
hostname: Mac-mini.local

2017-09-19 15:44:29,186 ERROR 16780 [-/::1/-/13ms GET /] nodejs.AssertionError [ERR_ASSERTION]: Can't find viewEngine for /Users/caigehui/Documents/ezz-example/app/view/dist/index.html
    at ContextView.[contextView#render] (/Users/caigehui/Documents/ezz-example/node_modules/egg-view/lib/context_view.js:71:5)
    at options.next (<anonymous>)
    at onFulfilled (/Users/caigehui/Documents/ezz-example/node_modules/co/index.js:65:19)
    at <anonymous>
generatedMessage: false
name: 'AssertionError [ERR_ASSERTION]'
code: 'ERR_ASSERTION'
actual: ''
expected: true
operator: '=='
pid: 16780
hostname: Mac-mini.local

2017-09-19 15:45:26,904 ERROR 16839 [-/::1/-/11ms GET /] nodejs.AssertionError [ERR_ASSERTION]: Can't find viewEngine for /Users/caigehui/Documents/ezz-example/app/view/dist/index.html
    at ContextView.[contextView#render] (/Users/caigehui/Documents/ezz-example/node_modules/egg-view/lib/context_view.js:71:5)
    at options.next (<anonymous>)
    at onFulfilled (/Users/caigehui/Documents/ezz-example/node_modules/co/index.js:65:19)
    at <anonymous>
generatedMessage: false
name: 'AssertionError [ERR_ASSERTION]'
code: 'ERR_ASSERTION'
actual: ''
expected: true
operator: '=='
pid: 16839
hostname: Mac-mini.local

2017-09-19 16:02:59,384 ERROR 17325 [-/::1/-/3ms GET /] nodejs.AssertionError [ERR_ASSERTION]: Can't find dist/index.html from /Users/caigehui/Documents/ezz-example/app/view
    at Map.resolve (/Users/caigehui/Documents/ezz-example/node_modules/egg-view/lib/view_manager.js:74:5)
    at resolve.next (<anonymous>)
    at onFulfilled (/Users/caigehui/Documents/ezz-example/node_modules/co/index.js:65:19)
    at <anonymous>
generatedMessage: false
name: 'AssertionError [ERR_ASSERTION]'
code: 'ERR_ASSERTION'
actual: undefined
expected: true
operator: '=='
pid: 17325
hostname: Mac-mini.local

2017-09-20 08:52:14,544 ERROR 1541 [-/127.0.0.1/-/6ms GET /] nodejs.ReferenceError: ctx is not defined
    at HomeController.index (/Users/caigehui/Documents/ezz-example/app/controller/home.js:6:13)
    at index.next (<anonymous>)
    at onFulfilled (/Users/caigehui/Documents/ezz-example/node_modules/co/index.js:65:19)
    at /Users/caigehui/Documents/ezz-example/node_modules/co/index.js:54:5
    at Promise (<anonymous>)
    at Object.co (/Users/caigehui/Documents/ezz-example/node_modules/co/index.js:50:10)
    at Object.toPromise (/Users/caigehui/Documents/ezz-example/node_modules/co/index.js:118:63)
    at next (/Users/caigehui/Documents/ezz-example/node_modules/co/index.js:99:29)
    at onFulfilled (/Users/caigehui/Documents/ezz-example/node_modules/co/index.js:69:7)
    at /Users/caigehui/Documents/ezz-example/node_modules/co/index.js:54:5

pid: 1541
hostname: Mac-mini.local

2017-09-20 08:52:54,124 ERROR 1730 [-/127.0.0.1/-/14ms GET /] nodejs.ReferenceError: ctx is not defined
    at HomeController.index (/Users/caigehui/Documents/ezz-example/app/controller/home.js:6:35)
    at index.next (<anonymous>)
    at onFulfilled (/Users/caigehui/Documents/ezz-example/node_modules/co/index.js:65:19)
    at /Users/caigehui/Documents/ezz-example/node_modules/co/index.js:54:5
    at Promise (<anonymous>)
    at Object.co (/Users/caigehui/Documents/ezz-example/node_modules/co/index.js:50:10)
    at Object.toPromise (/Users/caigehui/Documents/ezz-example/node_modules/co/index.js:118:63)
    at next (/Users/caigehui/Documents/ezz-example/node_modules/co/index.js:99:29)
    at onFulfilled (/Users/caigehui/Documents/ezz-example/node_modules/co/index.js:69:7)
    at /Users/caigehui/Documents/ezz-example/node_modules/co/index.js:54:5

pid: 1730
hostname: Mac-mini.local

2017-09-20 08:53:50,826 ERROR 1730 [-/127.0.0.1/-/4ms GET /] nodejs.ReferenceError: ctx is not defined
    at HomeController.index (/Users/caigehui/Documents/ezz-example/app/controller/home.js:6:35)
    at index.next (<anonymous>)
    at onFulfilled (/Users/caigehui/Documents/ezz-example/node_modules/co/index.js:65:19)
    at /Users/caigehui/Documents/ezz-example/node_modules/co/index.js:54:5
    at Promise (<anonymous>)
    at Object.co (/Users/caigehui/Documents/ezz-example/node_modules/co/index.js:50:10)
    at Object.toPromise (/Users/caigehui/Documents/ezz-example/node_modules/co/index.js:118:63)
    at next (/Users/caigehui/Documents/ezz-example/node_modules/co/index.js:99:29)
    at onFulfilled (/Users/caigehui/Documents/ezz-example/node_modules/co/index.js:69:7)
    at /Users/caigehui/Documents/ezz-example/node_modules/co/index.js:54:5

pid: 1730
hostname: Mac-mini.local

2017-09-20 08:54:23,011 ERROR 1787 [-/127.0.0.1/-/29ms GET /] nodejs.AssertionError [ERR_ASSERTION]: Can't find public/index.html from /Users/caigehui/Documents/ezz-example/app/view
    at Map.resolve (/Users/caigehui/Documents/ezz-example/node_modules/egg-view/lib/view_manager.js:74:5)
    at resolve.next (<anonymous>)
    at onFulfilled (/Users/caigehui/Documents/ezz-example/node_modules/co/index.js:65:19)
    at <anonymous>
generatedMessage: false
name: 'AssertionError [ERR_ASSERTION]'
code: 'ERR_ASSERTION'
actual: undefined
expected: true
operator: '=='
pid: 1787
hostname: Mac-mini.local

2017-09-26 08:45:31,587 ERROR 1315 [-/::1/-/9ms GET /] nodejs.ReferenceError: fs is not defined
    at HomeController.index (/Users/caigehui/Documents/ezz-example/app/controller/home.js:10:23)
    at index.next (<anonymous>)
    at onFulfilled (/Users/caigehui/Documents/ezz-example/node_modules/co/index.js:65:19)
    at /Users/caigehui/Documents/ezz-example/node_modules/co/index.js:54:5
    at Promise (<anonymous>)
    at Object.co (/Users/caigehui/Documents/ezz-example/node_modules/co/index.js:50:10)
    at Object.toPromise (/Users/caigehui/Documents/ezz-example/node_modules/co/index.js:118:63)
    at next (/Users/caigehui/Documents/ezz-example/node_modules/co/index.js:99:29)
    at onFulfilled (/Users/caigehui/Documents/ezz-example/node_modules/co/index.js:69:7)
    at /Users/caigehui/Documents/ezz-example/node_modules/co/index.js:54:5

pid: 1315
hostname: Mac-mini.local

2017-09-26 08:47:39,115 ERROR 1397 [-/::1/-/10ms GET /] nodejs.AssertionError [ERR_ASSERTION]: Can't find index.html from /Users/caigehui/Documents/ezz-example/app/view
    at Map.resolve (/Users/caigehui/Documents/ezz-example/node_modules/egg-view/lib/view_manager.js:74:5)
    at resolve.next (<anonymous>)
    at onFulfilled (/Users/caigehui/Documents/ezz-example/node_modules/co/index.js:65:19)
    at <anonymous>
generatedMessage: false
name: 'AssertionError [ERR_ASSERTION]'
code: 'ERR_ASSERTION'
actual: undefined
expected: true
operator: '=='
pid: 1397
hostname: Mac-mini.local

2017-09-26 11:04:06,198 ERROR 3783 [-/::1/-/232ms PUT /api/apiTest/hhhhh] nodejs.TypeError: Cannot read property 'length' of undefined
    at sjcl.cipher.aes (/Users/caigehui/Documents/ezz-example/node_modules/sjcl/sjcl.js:2:97)
    at Object.ia (/Users/caigehui/Documents/ezz-example/node_modules/sjcl/sjcl.js:55:467)
    at Object.decrypt (/Users/caigehui/Documents/ezz-example/node_modules/sjcl/sjcl.js:56:300)
    at ApiTestController.update (/Users/caigehui/Documents/ezz-example/app/controller/apiTest.js:12:31)
    at update.next (<anonymous>)
    at onFulfilled (/Users/caigehui/Documents/ezz-example/node_modules/co/index.js:65:19)
    at /Users/caigehui/Documents/ezz-example/node_modules/co/index.js:54:5
    at Promise (<anonymous>)
    at Object.co (/Users/caigehui/Documents/ezz-example/node_modules/co/index.js:50:10)
    at Object.toPromise (/Users/caigehui/Documents/ezz-example/node_modules/co/index.js:118:63)

pid: 3783
hostname: Mac-mini.local

2017-09-26 14:39:39,411 ERROR 8337 [-/::1/-/9ms GET /api/apsiTest/hhhhh] nodejs.ReferenceError: Must call super constructor in derived class before accessing 'this' or returning from derived constructor
    at NomatchController (/Users/caigehui/Documents/ezz-example/app/controller/nomatch.js:6:9)
    at Object.classControllerMiddleware (/Users/caigehui/Documents/ezz-example/node_modules/egg-core/lib/loader/mixin/controller.js:76:26)
    at classControllerMiddleware.next (<anonymous>)
    at Object.dispatch (/Users/caigehui/Documents/ezz-example/node_modules/egg-core/node_modules/koa-router/lib/router.js:334:12)
    at dispatch.next (<anonymous>)
    at onFulfilled (/Users/caigehui/Documents/ezz-example/node_modules/co/index.js:65:19)
    at /Users/caigehui/Documents/ezz-example/node_modules/co/index.js:54:5
    at Promise (<anonymous>)
    at Object.co (/Users/caigehui/Documents/ezz-example/node_modules/co/index.js:50:10)
    at Object.toPromise (/Users/caigehui/Documents/ezz-example/node_modules/co/index.js:118:63)

pid: 8337
hostname: Mac-mini.local

2017-09-26 14:43:59,565 ERROR 8474 [-/::1/-/10ms GET /api/apsiTest/hhhhh] nodejs.TypeError: Cannot set property 'status' of undefined
    at NomatchController (/Users/caigehui/Documents/ezz-example/app/controller/nomatch.js:6:26)
    at Object.classControllerMiddleware (/Users/caigehui/Documents/ezz-example/node_modules/egg-core/lib/loader/mixin/controller.js:76:26)
    at classControllerMiddleware.next (<anonymous>)
    at Object.dispatch (/Users/caigehui/Documents/ezz-example/node_modules/egg-core/node_modules/koa-router/lib/router.js:334:12)
    at dispatch.next (<anonymous>)
    at onFulfilled (/Users/caigehui/Documents/ezz-example/node_modules/co/index.js:65:19)
    at /Users/caigehui/Documents/ezz-example/node_modules/co/index.js:54:5
    at Promise (<anonymous>)
    at Object.co (/Users/caigehui/Documents/ezz-example/node_modules/co/index.js:50:10)
    at Object.toPromise (/Users/caigehui/Documents/ezz-example/node_modules/co/index.js:118:63)

pid: 8474
hostname: Mac-mini.local

2017-09-26 14:44:16,947 ERROR 8493 [-/::1/-/22ms GET /api/apsiTest/hhhhh] nodejs.TypeError: Cannot set property 'status' of undefined
    at NomatchController (/Users/caigehui/Documents/ezz-example/app/controller/nomatch.js:7:26)
    at Object.classControllerMiddleware (/Users/caigehui/Documents/ezz-example/node_modules/egg-core/lib/loader/mixin/controller.js:76:26)
    at classControllerMiddleware.next (<anonymous>)
    at Object.dispatch (/Users/caigehui/Documents/ezz-example/node_modules/egg-core/node_modules/koa-router/lib/router.js:334:12)
    at dispatch.next (<anonymous>)
    at onFulfilled (/Users/caigehui/Documents/ezz-example/node_modules/co/index.js:65:19)
    at /Users/caigehui/Documents/ezz-example/node_modules/co/index.js:54:5
    at Promise (<anonymous>)
    at Object.co (/Users/caigehui/Documents/ezz-example/node_modules/co/index.js:50:10)
    at Object.toPromise (/Users/caigehui/Documents/ezz-example/node_modules/co/index.js:118:63)

pid: 8493
hostname: Mac-mini.local

2017-09-26 14:51:17,983 ERROR 8662 [-/::1/-/133ms GET /api/apiTest/hhhhh] nodejs.Error: json decode: this isn't json!
    at Object.app.context.onerror (/Users/caigehui/Documents/ezz-example/node_modules/koa-onerror/index.js:36:24)
    at <anonymous>
    at process._tickCallback (internal/process/next_tick.js:188:7)

pid: 8662
hostname: Mac-mini.local

2017-09-26 14:51:43,046 ERROR 8662 [-/::1/-/1ms GET /api/apiTest/hhhhh] nodejs.Error: json decode: this isn't json!
    at Object.app.context.onerror (/Users/caigehui/Documents/ezz-example/node_modules/koa-onerror/index.js:36:24)
    at <anonymous>
    at process._tickCallback (internal/process/next_tick.js:188:7)

pid: 8662
hostname: Mac-mini.local

2017-09-26 15:00:42,908 ERROR 8982 [-/::1/-/166ms GET /api/apiTest/hhhhh] invalid {
  toString: [Function],
  message: 'json decode: this isn\'t json!' }
2017-09-26 16:55:08,449 ERROR 12275 [-/::1/-/7ms POST /api/login] nodejs.ReferenceError: ctx is not defined
    at LoginController.index (/Users/caigehui/Documents/ezz-example/app/controller/login.js:8:40)
    at index.next (<anonymous>)
    at onFulfilled (/Users/caigehui/Documents/ezz-example/node_modules/co/index.js:65:19)
    at /Users/caigehui/Documents/ezz-example/node_modules/co/index.js:54:5
    at Promise (<anonymous>)
    at Object.co (/Users/caigehui/Documents/ezz-example/node_modules/co/index.js:50:10)
    at Object.toPromise (/Users/caigehui/Documents/ezz-example/node_modules/co/index.js:118:63)
    at next (/Users/caigehui/Documents/ezz-example/node_modules/co/index.js:99:29)
    at onFulfilled (/Users/caigehui/Documents/ezz-example/node_modules/co/index.js:69:7)
    at /Users/caigehui/Documents/ezz-example/node_modules/co/index.js:54:5
    at Promise (<anonymous>)
    at Object.co (/Users/caigehui/Documents/ezz-example/node_modules/co/index.js:50:10)
    at Object.toPromise (/Users/caigehui/Documents/ezz-example/node_modules/co/index.js:118:63)
    at next (/Users/caigehui/Documents/ezz-example/node_modules/co/index.js:99:29)
    at onFulfilled (/Users/caigehui/Documents/ezz-example/node_modules/co/index.js:69:7)
    at /Users/caigehui/Documents/ezz-example/node_modules/co/index.js:54:5

pid: 12275
hostname: Mac-mini.local

2017-09-26 17:44:12,180 ERROR 13557 [-/::1/-/53ms POST /api/login] nodejs.ReferenceError: ctx is not defined
    at LoginController.index (/Users/caigehui/Documents/ezz-example/app/controller/login.js:9:9)
    at index.next (<anonymous>)
    at onFulfilled (/Users/caigehui/Documents/ezz-example/node_modules/co/index.js:65:19)
    at <anonymous>
    at process._tickCallback (internal/process/next_tick.js:188:7)

pid: 13557
hostname: Mac-mini.local

2017-09-27 09:01:41,510 ERROR 1666 [egg-mongoose] mongodb://localhost:27017/ezz disconnected
2017-09-27 09:01:41,513 ERROR 1666 nodejs.MongoError: [egg-mongoose]failed to connect to server [localhost:27017] on first connect [MongoError: connect ECONNREFUSED 127.0.0.1:27017]
    at Pool.<anonymous> (/Users/caigehui/Documents/ezz-example/node_modules/mongodb-core/lib/topologies/server.js:336:35)
    at emitOne (events.js:115:13)
    at Pool.emit (events.js:210:7)
    at Connection.<anonymous> (/Users/caigehui/Documents/ezz-example/node_modules/mongodb-core/lib/connection/pool.js:280:12)
    at Object.onceWrapper (events.js:318:30)
    at emitTwo (events.js:125:13)
    at Connection.emit (events.js:213:7)
    at Socket.<anonymous> (/Users/caigehui/Documents/ezz-example/node_modules/mongodb-core/lib/connection/connection.js:187:49)
    at Object.onceWrapper (events.js:316:30)
    at emitOne (events.js:115:13)
    at Socket.emit (events.js:210:7)
    at emitErrorNT (internal/streams/destroy.js:64:8)
    at _combinedTickCallback (internal/process/next_tick.js:138:11)
    at process._tickCallback (internal/process/next_tick.js:180:9)
name: 'MongoError'
message: '[egg-mongoose]failed to connect to server [localhost:27017] on first connect [MongoError: connect ECONNREFUSED 127.0.0.1:27017]'
pid: 1666
hostname: Mac-mini.local

2017-09-27 09:16:41,904 ERROR 2050 [-/::1/-/16ms GET /api/users] nodejs.TypeError: Cannot set property 'body' of undefined
    at Object.<anonymous> (/Users/caigehui/Documents/ezz-example/app/middleware/save_session.js:8:19)
    at Generator.next (<anonymous>)
    at onFulfilled (/Users/caigehui/Documents/ezz-example/node_modules/co/index.js:65:19)
    at <anonymous>
    at process._tickCallback (internal/process/next_tick.js:188:7)

pid: 2050
hostname: Mac-mini.local

2017-09-28 16:28:48,609 ERROR 16784 [egg-mongoose] mongodb://localhost:27017/ezz disconnected
2017-09-28 16:28:48,612 ERROR 16784 nodejs.MongoError: [egg-mongoose]failed to connect to server [localhost:27017] on first connect [MongoError: connect ECONNREFUSED 127.0.0.1:27017]
    at Pool.<anonymous> (/Users/caigehui/Documents/ezz-example/node_modules/mongodb-core/lib/topologies/server.js:336:35)
    at emitOne (events.js:115:13)
    at Pool.emit (events.js:210:7)
    at Connection.<anonymous> (/Users/caigehui/Documents/ezz-example/node_modules/mongodb-core/lib/connection/pool.js:280:12)
    at Object.onceWrapper (events.js:318:30)
    at emitTwo (events.js:125:13)
    at Connection.emit (events.js:213:7)
    at Socket.<anonymous> (/Users/caigehui/Documents/ezz-example/node_modules/mongodb-core/lib/connection/connection.js:187:49)
    at Object.onceWrapper (events.js:316:30)
    at emitOne (events.js:115:13)
    at Socket.emit (events.js:210:7)
    at emitErrorNT (internal/streams/destroy.js:64:8)
    at _combinedTickCallback (internal/process/next_tick.js:138:11)
    at process._tickCallback (internal/process/next_tick.js:180:9)
name: 'MongoError'
message: '[egg-mongoose]failed to connect to server [localhost:27017] on first connect [MongoError: connect ECONNREFUSED 127.0.0.1:27017]'
pid: 16784
hostname: Mac-mini.local

2017-09-29 15:29:43,540 ERROR 12432 [-/127.0.0.1/-/20ms POST /api/files] nodejs.ReferenceError: stream is not defined
    at FilesController.create (/Users/caigehui/Documents/ezz-example/app/controller/files.js:39:38)
    at create.throw (<anonymous>)
    at onRejected (/Users/caigehui/Documents/ezz-example/node_modules/co/index.js:81:24)
    at next (/Users/caigehui/Documents/ezz-example/node_modules/co/index.js:101:14)
    at onFulfilled (/Users/caigehui/Documents/ezz-example/node_modules/co/index.js:69:7)
    at <anonymous>
    at process._tickCallback (internal/process/next_tick.js:188:7)

pid: 12432
hostname: Mac-mini.local

2017-09-29 15:37:50,501 ERROR 12615 [-/127.0.0.1/-/21ms POST /api/files] nodejs.ReferenceError: stream is not defined
    at FilesController.create (/Users/caigehui/Documents/ezz-example/app/controller/files.js:42:38)
    at create.next (<anonymous>)
    at onFulfilled (/Users/caigehui/Documents/ezz-example/node_modules/co/index.js:65:19)
    at <anonymous>
    at process._tickCallback (internal/process/next_tick.js:188:7)

pid: 12615
hostname: Mac-mini.local

2017-09-29 15:38:41,007 ERROR 12639 [-/127.0.0.1/-/21ms POST /api/files] nodejs.ReferenceError: stream is not defined
    at FilesController.create (/Users/caigehui/Documents/ezz-example/app/controller/files.js:42:38)
    at create.next (<anonymous>)
    at onFulfilled (/Users/caigehui/Documents/ezz-example/node_modules/co/index.js:65:19)
    at <anonymous>
    at process._tickCallback (internal/process/next_tick.js:188:7)

pid: 12639
hostname: Mac-mini.local

2017-09-29 15:39:42,357 ERROR 12676 [-/127.0.0.1/-/28ms POST /api/files] nodejs.TypeError: part.write is not a function
    at FilesController.create (/Users/caigehui/Documents/ezz-example/app/controller/files.js:38:39)
    at create.next (<anonymous>)
    at onFulfilled (/Users/caigehui/Documents/ezz-example/node_modules/co/index.js:65:19)
    at <anonymous>
    at process._tickCallback (internal/process/next_tick.js:188:7)

pid: 12676
hostname: Mac-mini.local

2017-09-29 16:45:29,776 ERROR 14953 [-/127.0.0.1/-/18ms POST /api/files] nodejs.TypeError: Cannot read property 'getUploadDirPath' of undefined
    at FilesController.create (/Users/caigehui/Documents/ezz-example/app/controller/files.js:35:54)
    at create.next (<anonymous>)
    at onFulfilled (/Users/caigehui/Documents/ezz-example/node_modules/co/index.js:65:19)
    at <anonymous>
    at process._tickCallback (internal/process/next_tick.js:188:7)

pid: 14953
hostname: Mac-mini.local

2017-09-29 16:45:42,665 ERROR 14978 [-/127.0.0.1/-/22ms POST /api/files] nodejs.TypeError: Cannot read property 'getUploadPath' of undefined
    at FilesController.create (/Users/caigehui/Documents/ezz-example/app/controller/files.js:35:54)
    at create.next (<anonymous>)
    at onFulfilled (/Users/caigehui/Documents/ezz-example/node_modules/co/index.js:65:19)
    at <anonymous>
    at process._tickCallback (internal/process/next_tick.js:188:7)

pid: 14978
hostname: Mac-mini.local

2017-09-29 16:47:24,355 ERROR 15078 [-/127.0.0.1/-/18ms POST /api/files] nodejs.TypeError: You may only yield a function, promise, generator, array, or object, but the following object was passed: "[object Object]"
    at next (/Users/caigehui/Documents/ezz-example/node_modules/co/index.js:101:25)
    at onFulfilled (/Users/caigehui/Documents/ezz-example/node_modules/co/index.js:69:7)
    at <anonymous>
    at process._tickCallback (internal/process/next_tick.js:188:7)

pid: 15078
hostname: Mac-mini.local

2017-09-30 12:03:59,221 ERROR 3936 [-/::1/-/18ms POST /api/files?_csrf=CU4Uif8Xx5KtRhDhYJAB2nx1] nodejs.TypeError: Cannot read property 'bind' of undefined
    at Promise (/Users/caigehui/Documents/ezz-example/app/extend/helper.js:13:38)
    at Promise (<anonymous>)
    at Helper.getFileHash (/Users/caigehui/Documents/ezz-example/app/extend/helper.js:11:16)
    at FilesController.create (/Users/caigehui/Documents/ezz-example/app/controller/base/files.js:20:34)
    at create.next (<anonymous>)
    at onFulfilled (/Users/caigehui/Documents/ezz-example/node_modules/co/index.js:65:19)
    at <anonymous>
    at process._tickCallback (internal/process/next_tick.js:188:7)

pid: 3936
hostname: Mac-mini.local

2017-09-30 12:05:29,875 ERROR 4022 nodejs.TypeError: Cannot read property 'update' of undefined (uncaughtException throw 1 times on pid:4022)
    at FileStream.update (crypto.js:97:15)
    at emitOne (events.js:115:13)
    at FileStream.emit (events.js:210:7)
    at FileStream.Readable.read (_stream_readable.js:478:10)
    at flow (_stream_readable.js:849:34)
    at resume_ (_stream_readable.js:831:3)
    at _combinedTickCallback (internal/process/next_tick.js:138:11)
    at process._tickCallback (internal/process/next_tick.js:180:9)

pid: 4022
hostname: Mac-mini.local

2017-09-30 12:05:29,882 ERROR 4022 [-/undefined/-/29ms POST /api/files?_csrf=CU4Uif8Xx5KtRhDhYJAB2nx1] nodejs.HPE_CB_bodyError: Parse Error
HPE_CB_bodyError: Parse Error
bytesParsed: 0
code: 'HPE_CB_body'
headerSent: true
name: 'HPE_CB_bodyError'
pid: 4022
hostname: Mac-mini.local

2017-09-30 13:08:11,359 ERROR 5281 [-/::1/-/24ms POST /api/files?_csrf=CU4Uif8Xx5KtRhDhYJAB2nx1] nodejs.ReferenceError: part is not defined
    at FilesController.create (/Users/caigehui/Documents/ezz-example/app/controller/base/files.js:13:4)
    at create.next (<anonymous>)
    at onFulfilled (/Users/caigehui/Documents/ezz-example/node_modules/co/index.js:65:19)
    at <anonymous>
    at process._tickCallback (internal/process/next_tick.js:188:7)

pid: 5281
hostname: Mac-mini.local

2017-09-30 13:08:44,124 ERROR 5352 [-/::1/-/23ms POST /api/files?_csrf=CU4Uif8Xx5KtRhDhYJAB2nx1] nodejs.ReferenceError: part is not defined
    at FilesController.create (/Users/caigehui/Documents/ezz-example/app/controller/base/files.js:13:4)
    at create.next (<anonymous>)
    at onFulfilled (/Users/caigehui/Documents/ezz-example/node_modules/co/index.js:65:19)
    at <anonymous>
    at process._tickCallback (internal/process/next_tick.js:188:7)

pid: 5352
hostname: Mac-mini.local

2017-09-30 13:09:04,436 ERROR 5378 [-/::1/-/26ms POST /api/files?_csrf=CU4Uif8Xx5KtRhDhYJAB2nx1] nodejs.TypeError: Cannot read property 'tolowercase' of undefined
    at Helper.isImage (/Users/caigehui/Documents/ezz-example/app/extend/helper.js:32:57)
    at FilesController.create (/Users/caigehui/Documents/ezz-example/app/controller/base/files.js:18:32)
    at create.next (<anonymous>)
    at onFulfilled (/Users/caigehui/Documents/ezz-example/node_modules/co/index.js:65:19)
    at <anonymous>
    at process._tickCallback (internal/process/next_tick.js:188:7)

pid: 5378
hostname: Mac-mini.local

2017-09-30 13:09:35,174 ERROR 5426 [-/::1/-/24ms POST /api/files?_csrf=CU4Uif8Xx5KtRhDhYJAB2nx1] nodejs.TypeError: rs.on is not a function
    at Promise (/Users/caigehui/Documents/ezz-example/app/extend/helper.js:21:16)
    at Promise (<anonymous>)
    at Helper.getFileHash (/Users/caigehui/Documents/ezz-example/app/extend/helper.js:19:16)
    at FilesController.create (/Users/caigehui/Documents/ezz-example/app/controller/base/files.js:17:35)
    at create.next (<anonymous>)
    at onFulfilled (/Users/caigehui/Documents/ezz-example/node_modules/co/index.js:65:19)
    at <anonymous>
    at process._tickCallback (internal/process/next_tick.js:188:7)

pid: 5426
hostname: Mac-mini.local

2017-09-30 13:10:35,730 ERROR 5457 [-/::1/-/23ms POST /api/files?_csrf=CU4Uif8Xx5KtRhDhYJAB2nx1] nodejs.TypeError: rs.on is not a function
    at Promise (/Users/caigehui/Documents/ezz-example/app/extend/helper.js:21:16)
    at Promise (<anonymous>)
    at Helper.getFileHash (/Users/caigehui/Documents/ezz-example/app/extend/helper.js:19:16)
    at FilesController.create (/Users/caigehui/Documents/ezz-example/app/controller/base/files.js:17:35)
    at create.next (<anonymous>)
    at onFulfilled (/Users/caigehui/Documents/ezz-example/node_modules/co/index.js:65:19)
    at <anonymous>
    at process._tickCallback (internal/process/next_tick.js:188:7)

pid: 5457
hostname: Mac-mini.local

2017-09-30 13:11:18,978 ERROR 5512 [-/::1/-/24ms POST /api/files?_csrf=CU4Uif8Xx5KtRhDhYJAB2nx1] nodejs.TypeError: rs.on is not a function
    at Promise (/Users/caigehui/Documents/ezz-example/app/extend/helper.js:21:16)
    at Promise (<anonymous>)
    at Helper.getFileHash (/Users/caigehui/Documents/ezz-example/app/extend/helper.js:19:16)
    at FilesController.create (/Users/caigehui/Documents/ezz-example/app/controller/base/files.js:17:35)
    at create.next (<anonymous>)
    at onFulfilled (/Users/caigehui/Documents/ezz-example/node_modules/co/index.js:65:19)
    at <anonymous>
    at process._tickCallback (internal/process/next_tick.js:188:7)

pid: 5512
hostname: Mac-mini.local

2017-09-30 13:12:14,666 ERROR 5583 [-/::1/-/25ms POST /api/files?_csrf=CU4Uif8Xx5KtRhDhYJAB2nx1] nodejs.TypeError: filename.tolowercase is not a function
    at Helper.isImage (/Users/caigehui/Documents/ezz-example/app/extend/helper.js:32:58)
    at FilesController.create (/Users/caigehui/Documents/ezz-example/app/controller/base/files.js:18:32)
    at create.next (<anonymous>)
    at onFulfilled (/Users/caigehui/Documents/ezz-example/node_modules/co/index.js:65:19)
    at <anonymous>
    at process._tickCallback (internal/process/next_tick.js:188:7)

pid: 5583
hostname: Mac-mini.local

2017-09-30 14:04:30,683 ERROR 6595 [-/::1/-/31ms POST /api/files?_csrf=CU4Uif8Xx5KtRhDhYJAB2nx1] nodejs.TypeError: You may only yield a function, promise, generator, array, or object, but the following object was passed: "undefined"
    at next (/Users/caigehui/Documents/ezz-example/node_modules/co/index.js:101:25)
    at onFulfilled (/Users/caigehui/Documents/ezz-example/node_modules/co/index.js:69:7)
    at <anonymous>
    at process._tickCallback (internal/process/next_tick.js:188:7)

pid: 6595
hostname: Mac-mini.local

2017-09-30 14:29:28,128 ERROR 6967 [-/::1/-/35ms POST /api/files?_csrf=CU4Uif8Xx5KtRhDhYJAB2nx1] nodejs.TypeError: Cannot read property 'getUploadPath' of undefined
    at FilesController.create (/Users/caigehui/Documents/ezz-example/app/controller/base/files.js:19:20)
    at create.next (<anonymous>)
    at onFulfilled (/Users/caigehui/Documents/ezz-example/node_modules/co/index.js:65:19)
    at <anonymous>
    at process._tickCallback (internal/process/next_tick.js:188:7)

pid: 6967
hostname: Mac-mini.local

2017-09-30 15:54:13,753 ERROR 9399 [-/::1/-/32ms POST /api/files?_csrf=CU4Uif8Xx5KtRhDhYJAB2nx1] nodejs.ReferenceError: size is not defined
    at FilesController.create (/Users/caigehui/Documents/ezz-example/app/controller/base/files.js:20:23)
    at create.next (<anonymous>)
    at onFulfilled (/Users/caigehui/Documents/ezz-example/node_modules/co/index.js:65:19)
    at <anonymous>
    at process._tickCallback (internal/process/next_tick.js:188:7)

pid: 9399
hostname: Mac-mini.local

2017-09-30 15:57:13,012 ERROR 9521 [-/::1/-/39ms POST /api/files?_csrf=CU4Uif8Xx5KtRhDhYJAB2nx1] nodejs.Error: gm().stream() or gm().write() with a non-readable stream.
    at gm._spawn (/Users/caigehui/Documents/ezz-example/node_modules/gm/lib/command.js:258:15)
    at gm._exec (/Users/caigehui/Documents/ezz-example/node_modules/gm/lib/command.js:190:17)
    at gm.proto.(anonymous function) [as size] (/Users/caigehui/Documents/ezz-example/node_modules/gm/lib/getters.js:68:12)
    at Promise (/Users/caigehui/Documents/ezz-example/app/extend/helper.js:51:18)
    at Promise (<anonymous>)
    at Helper.getImageSize (/Users/caigehui/Documents/ezz-example/app/extend/helper.js:49:16)
    at FilesController.create (/Users/caigehui/Documents/ezz-example/app/controller/base/files.js:18:39)
    at create.next (<anonymous>)
    at onFulfilled (/Users/caigehui/Documents/ezz-example/node_modules/co/index.js:65:19)
    at <anonymous>
    at process._tickCallback (internal/process/next_tick.js:188:7)

pid: 9521
hostname: Mac-mini.local

2017-09-30 16:03:00,246 ERROR 9721 [-/::1/-/34ms POST /api/files?_csrf=CU4Uif8Xx5KtRhDhYJAB2nx1] nodejs.Error: gm().stream() or gm().write() with a non-readable stream.
    at gm._spawn (/Users/caigehui/Documents/ezz-example/node_modules/gm/lib/command.js:258:15)
    at gm._exec (/Users/caigehui/Documents/ezz-example/node_modules/gm/lib/command.js:190:17)
    at gm.proto.(anonymous function) [as size] (/Users/caigehui/Documents/ezz-example/node_modules/gm/lib/getters.js:68:12)
    at Promise (/Users/caigehui/Documents/ezz-example/app/extend/helper.js:52:18)
    at Promise (<anonymous>)
    at Helper.getImageSize (/Users/caigehui/Documents/ezz-example/app/extend/helper.js:50:16)
    at FilesController.create (/Users/caigehui/Documents/ezz-example/app/controller/base/files.js:18:39)
    at create.next (<anonymous>)
    at onFulfilled (/Users/caigehui/Documents/ezz-example/node_modules/co/index.js:65:19)
    at <anonymous>
    at process._tickCallback (internal/process/next_tick.js:188:7)

pid: 9721
hostname: Mac-mini.local

2017-09-30 16:06:09,304 ERROR 9866 [-/::1/-/34ms POST /api/files?_csrf=CU4Uif8Xx5KtRhDhYJAB2nx1] nodejs.Error: gm().stream() or gm().write() with a non-readable stream.
    at gm._spawn (/Users/caigehui/Documents/ezz-example/node_modules/gm/lib/command.js:258:15)
    at gm._exec (/Users/caigehui/Documents/ezz-example/node_modules/gm/lib/command.js:190:17)
    at gm.proto.(anonymous function) [as size] (/Users/caigehui/Documents/ezz-example/node_modules/gm/lib/getters.js:68:12)
    at Promise (/Users/caigehui/Documents/ezz-example/app/extend/helper.js:52:18)
    at Promise (<anonymous>)
    at Helper.getImageSize (/Users/caigehui/Documents/ezz-example/app/extend/helper.js:50:16)
    at FilesController.create (/Users/caigehui/Documents/ezz-example/app/controller/base/files.js:18:39)
    at create.next (<anonymous>)
    at onFulfilled (/Users/caigehui/Documents/ezz-example/node_modules/co/index.js:65:19)
    at <anonymous>
    at process._tickCallback (internal/process/next_tick.js:188:7)

pid: 9866
hostname: Mac-mini.local

2017-09-30 16:06:36,908 ERROR 9899 [-/::1/-/40ms POST /api/files?_csrf=CU4Uif8Xx5KtRhDhYJAB2nx1] nodejs.Error: gm().stream() or gm().write() with a non-readable stream.
    at gm._spawn (/Users/caigehui/Documents/ezz-example/node_modules/gm/lib/command.js:258:15)
    at gm._exec (/Users/caigehui/Documents/ezz-example/node_modules/gm/lib/command.js:190:17)
    at gm.proto.(anonymous function) [as size] (/Users/caigehui/Documents/ezz-example/node_modules/gm/lib/getters.js:68:12)
    at Promise (/Users/caigehui/Documents/ezz-example/app/extend/helper.js:53:18)
    at Promise (<anonymous>)
    at Helper.getImageSize (/Users/caigehui/Documents/ezz-example/app/extend/helper.js:50:16)
    at FilesController.create (/Users/caigehui/Documents/ezz-example/app/controller/base/files.js:18:39)
    at create.next (<anonymous>)
    at onFulfilled (/Users/caigehui/Documents/ezz-example/node_modules/co/index.js:65:19)
    at <anonymous>
    at process._tickCallback (internal/process/next_tick.js:188:7)

pid: 9899
hostname: Mac-mini.local

2017-09-30 16:07:00,509 ERROR 9933 [-/::1/-/41ms POST /api/files?_csrf=CU4Uif8Xx5KtRhDhYJAB2nx1] nodejs.Error: gm().stream() or gm().write() with a non-readable stream.
    at gm._spawn (/Users/caigehui/Documents/ezz-example/node_modules/gm/lib/command.js:258:15)
    at gm._exec (/Users/caigehui/Documents/ezz-example/node_modules/gm/lib/command.js:190:17)
    at gm.proto.(anonymous function) [as size] (/Users/caigehui/Documents/ezz-example/node_modules/gm/lib/getters.js:68:12)
    at Promise (/Users/caigehui/Documents/ezz-example/app/extend/helper.js:53:18)
    at Promise (<anonymous>)
    at Helper.getImageSize (/Users/caigehui/Documents/ezz-example/app/extend/helper.js:50:16)
    at FilesController.create (/Users/caigehui/Documents/ezz-example/app/controller/base/files.js:18:39)
    at create.next (<anonymous>)
    at onFulfilled (/Users/caigehui/Documents/ezz-example/node_modules/co/index.js:65:19)
    at <anonymous>
    at process._tickCallback (internal/process/next_tick.js:188:7)

pid: 9933
hostname: Mac-mini.local

2017-09-30 16:08:06,862 ERROR 9960 nodejs.TypeError: rs.end is not a function (uncaughtException throw 1 times on pid:9960)
    at FileStream.rs.on (/Users/caigehui/Documents/ezz-example/app/extend/helper.js:24:20)
    at emitNone (events.js:110:20)
    at FileStream.emit (events.js:207:7)
    at endReadableNT (_stream_readable.js:1059:12)
    at _combinedTickCallback (internal/process/next_tick.js:138:11)
    at process._tickCallback (internal/process/next_tick.js:180:9)

pid: 9960
hostname: Mac-mini.local

2017-09-30 16:08:37,014 ERROR 10023 [-/::1/-/116ms POST /api/files?_csrf=CU4Uif8Xx5KtRhDhYJAB2nx1] nodejs.Error: gm().stream() or gm().write() with a non-readable stream.
    at gm._spawn (/Users/caigehui/Documents/ezz-example/node_modules/gm/lib/command.js:258:15)
    at gm._exec (/Users/caigehui/Documents/ezz-example/node_modules/gm/lib/command.js:190:17)
    at gm.proto.(anonymous function) [as size] (/Users/caigehui/Documents/ezz-example/node_modules/gm/lib/getters.js:68:12)
    at Promise (/Users/caigehui/Documents/ezz-example/app/extend/helper.js:52:18)
    at Promise (<anonymous>)
    at Helper.getImageSize (/Users/caigehui/Documents/ezz-example/app/extend/helper.js:49:16)
    at FilesController.create (/Users/caigehui/Documents/ezz-example/app/controller/base/files.js:18:39)
    at create.next (<anonymous>)
    at onFulfilled (/Users/caigehui/Documents/ezz-example/node_modules/co/index.js:65:19)
    at <anonymous>
    at process._tickCallback (internal/process/next_tick.js:188:7)

pid: 10023
hostname: Mac-mini.local

2017-09-30 16:13:02,776 ERROR 10172 [-/::1/-/88ms POST /api/files?_csrf=CU4Uif8Xx5KtRhDhYJAB2nx1] nodejs.Error: gm().stream() or gm().write() with a non-readable stream.
    at gm._spawn (/Users/caigehui/Documents/ezz-example/node_modules/gm/lib/command.js:258:15)
    at gm._exec (/Users/caigehui/Documents/ezz-example/node_modules/gm/lib/command.js:190:17)
    at gm.proto.(anonymous function) [as size] (/Users/caigehui/Documents/ezz-example/node_modules/gm/lib/getters.js:68:12)
    at Promise (/Users/caigehui/Documents/ezz-example/app/extend/helper.js:51:18)
    at Promise (<anonymous>)
    at Helper.getImageSize (/Users/caigehui/Documents/ezz-example/app/extend/helper.js:49:16)
    at FilesController.create (/Users/caigehui/Documents/ezz-example/app/controller/base/files.js:19:22)
    at create.next (<anonymous>)
    at onFulfilled (/Users/caigehui/Documents/ezz-example/node_modules/co/index.js:65:19)
    at <anonymous>

pid: 10172
hostname: Mac-mini.local

2017-09-30 16:13:40,051 ERROR 10215 [-/::1/-/77ms POST /api/files?_csrf=CU4Uif8Xx5KtRhDhYJAB2nx1] nodejs.Error: gm().stream() or gm().write() with a non-readable stream.
    at gm._spawn (/Users/caigehui/Documents/ezz-example/node_modules/gm/lib/command.js:258:15)
    at gm._exec (/Users/caigehui/Documents/ezz-example/node_modules/gm/lib/command.js:190:17)
    at gm.proto.(anonymous function) [as size] (/Users/caigehui/Documents/ezz-example/node_modules/gm/lib/getters.js:68:12)
    at Promise (/Users/caigehui/Documents/ezz-example/app/extend/helper.js:51:18)
    at Promise (<anonymous>)
    at Helper.getImageSize (/Users/caigehui/Documents/ezz-example/app/extend/helper.js:49:16)
    at FilesController.create (/Users/caigehui/Documents/ezz-example/app/controller/base/files.js:19:22)
    at create.next (<anonymous>)
    at onFulfilled (/Users/caigehui/Documents/ezz-example/node_modules/co/index.js:65:19)
    at <anonymous>

pid: 10215
hostname: Mac-mini.local

2017-09-30 16:38:03,719 ERROR 11073 [-/::1/-/11ms GET /api/files] nodejs.TypeError: Cannot read property 'getUploadPath' of undefined
    at FilesController.index (/Users/caigehui/Documents/ezz-example/app/controller/base/files.js:6:30)
    at index.next (<anonymous>)
    at onFulfilled (/Users/caigehui/Documents/ezz-example/node_modules/co/index.js:65:19)
    at /Users/caigehui/Documents/ezz-example/node_modules/co/index.js:54:5
    at Promise (<anonymous>)
    at Object.co (/Users/caigehui/Documents/ezz-example/node_modules/co/index.js:50:10)
    at Object.toPromise (/Users/caigehui/Documents/ezz-example/node_modules/co/index.js:118:63)
    at next (/Users/caigehui/Documents/ezz-example/node_modules/co/index.js:99:29)
    at onFulfilled (/Users/caigehui/Documents/ezz-example/node_modules/co/index.js:69:7)
    at /Users/caigehui/Documents/ezz-example/node_modules/co/index.js:54:5
    at Promise (<anonymous>)
    at Object.co (/Users/caigehui/Documents/ezz-example/node_modules/co/index.js:50:10)
    at Object.toPromise (/Users/caigehui/Documents/ezz-example/node_modules/co/index.js:118:63)
    at next (/Users/caigehui/Documents/ezz-example/node_modules/co/index.js:99:29)
    at onFulfilled (/Users/caigehui/Documents/ezz-example/node_modules/co/index.js:69:7)
    at /Users/caigehui/Documents/ezz-example/node_modules/co/index.js:54:5

pid: 11073
hostname: Mac-mini.local

2017-09-30 16:38:14,733 ERROR 11073 [-/::1/-/2ms GET /api/files] nodejs.TypeError: Cannot read property 'getUploadPath' of undefined
    at FilesController.index (/Users/caigehui/Documents/ezz-example/app/controller/base/files.js:6:30)
    at index.next (<anonymous>)
    at onFulfilled (/Users/caigehui/Documents/ezz-example/node_modules/co/index.js:65:19)
    at /Users/caigehui/Documents/ezz-example/node_modules/co/index.js:54:5
    at Promise (<anonymous>)
    at Object.co (/Users/caigehui/Documents/ezz-example/node_modules/co/index.js:50:10)
    at Object.toPromise (/Users/caigehui/Documents/ezz-example/node_modules/co/index.js:118:63)
    at next (/Users/caigehui/Documents/ezz-example/node_modules/co/index.js:99:29)
    at onFulfilled (/Users/caigehui/Documents/ezz-example/node_modules/co/index.js:69:7)
    at /Users/caigehui/Documents/ezz-example/node_modules/co/index.js:54:5
    at Promise (<anonymous>)
    at Object.co (/Users/caigehui/Documents/ezz-example/node_modules/co/index.js:50:10)
    at Object.toPromise (/Users/caigehui/Documents/ezz-example/node_modules/co/index.js:118:63)
    at next (/Users/caigehui/Documents/ezz-example/node_modules/co/index.js:99:29)
    at onFulfilled (/Users/caigehui/Documents/ezz-example/node_modules/co/index.js:69:7)
    at /Users/caigehui/Documents/ezz-example/node_modules/co/index.js:54:5

pid: 11073
hostname: Mac-mini.local

2017-09-30 17:01:55,694 ERROR 11900 [-/::1/-/24ms POST /api/files?_csrf=CU4Uif8Xx5KtRhDhYJAB2nx1] nodejs.TypeError: Cannot read property 'createFile' of undefined
    at FilesController.create (/Users/caigehui/Documents/ezz-example/app/controller/base/files.js:25:27)
    at create.next (<anonymous>)
    at onFulfilled (/Users/caigehui/Documents/ezz-example/node_modules/co/index.js:65:19)
    at <anonymous>
    at process._tickCallback (internal/process/next_tick.js:188:7)

pid: 11900
hostname: Mac-mini.local

2017-09-30 17:04:06,356 ERROR 11975 [-/::1/-/23ms POST /api/files?_csrf=CU4Uif8Xx5KtRhDhYJAB2nx1] nodejs.TypeError: Cannot read property 'createFile' of undefined
    at FilesController.create (/Users/caigehui/Documents/ezz-example/app/controller/base/files.js:25:27)
    at create.next (<anonymous>)
    at onFulfilled (/Users/caigehui/Documents/ezz-example/node_modules/co/index.js:65:19)
    at <anonymous>
    at process._tickCallback (internal/process/next_tick.js:188:7)

pid: 11975
hostname: Mac-mini.local

2017-09-30 17:25:56,572 ERROR 12460 [-/::1/-/27ms POST /api/files?_csrf=CU4Uif8Xx5KtRhDhYJAB2nx1] nodejs.TypeError: this.config.getTempPath is not a function
    at FilesController.create (/Users/caigehui/Documents/ezz-example/app/controller/base/files.js:17:34)
    at create.next (<anonymous>)
    at onFulfilled (/Users/caigehui/Documents/ezz-example/node_modules/co/index.js:65:19)
    at <anonymous>
    at process._tickCallback (internal/process/next_tick.js:188:7)

pid: 12460
hostname: Mac-mini.local

2017-10-02 12:15:34,613 ERROR 958 [egg-mongoose] mongodb://localhost:27017/ezz disconnected
2017-10-02 12:15:34,616 ERROR 958 nodejs.MongoError: [egg-mongoose]failed to connect to server [localhost:27017] on first connect [MongoError: connect ECONNREFUSED 127.0.0.1:27017]
    at Pool.<anonymous> (/Users/caigehui/Documents/ezz-example/node_modules/mongodb-core/lib/topologies/server.js:336:35)
    at emitOne (events.js:115:13)
    at Pool.emit (events.js:210:7)
    at Connection.<anonymous> (/Users/caigehui/Documents/ezz-example/node_modules/mongodb-core/lib/connection/pool.js:280:12)
    at Object.onceWrapper (events.js:318:30)
    at emitTwo (events.js:125:13)
    at Connection.emit (events.js:213:7)
    at Socket.<anonymous> (/Users/caigehui/Documents/ezz-example/node_modules/mongodb-core/lib/connection/connection.js:187:49)
    at Object.onceWrapper (events.js:316:30)
    at emitOne (events.js:115:13)
    at Socket.emit (events.js:210:7)
    at emitErrorNT (internal/streams/destroy.js:64:8)
    at _combinedTickCallback (internal/process/next_tick.js:138:11)
    at process._tickCallback (internal/process/next_tick.js:180:9)
name: 'MongoError'
message: '[egg-mongoose]failed to connect to server [localhost:27017] on first connect [MongoError: connect ECONNREFUSED 127.0.0.1:27017]'
pid: 958
hostname: Mac-mini.local

2017-10-02 14:53:01,086 ERROR 2560 [-/127.0.0.1/-/17ms POST /api/files] nodejs.ReferenceError: filename is not defined
    at FilesController.create (/Users/caigehui/Documents/ezz-example/app/controller/base/files.js:20:17)
    at create.next (<anonymous>)
    at onFulfilled (/Users/caigehui/Documents/ezz-example/node_modules/co/index.js:65:19)
    at <anonymous>
    at process._tickCallback (internal/process/next_tick.js:188:7)

pid: 2560
hostname: Mac-mini.local

2017-10-09 09:53:47,511 ERROR 2230 [egg-mongoose] mongodb://localhost:27017/ezz disconnected
2017-10-09 09:53:47,515 ERROR 2230 nodejs.MongoError: [egg-mongoose]failed to connect to server [localhost:27017] on first connect [MongoError: connect ECONNREFUSED 127.0.0.1:27017]
    at Pool.<anonymous> (/Users/caigehui/Documents/ezz-example/node_modules/mongodb-core/lib/topologies/server.js:336:35)
    at emitOne (events.js:115:13)
    at Pool.emit (events.js:210:7)
    at Connection.<anonymous> (/Users/caigehui/Documents/ezz-example/node_modules/mongodb-core/lib/connection/pool.js:280:12)
    at Object.onceWrapper (events.js:318:30)
    at emitTwo (events.js:125:13)
    at Connection.emit (events.js:213:7)
    at Socket.<anonymous> (/Users/caigehui/Documents/ezz-example/node_modules/mongodb-core/lib/connection/connection.js:187:49)
    at Object.onceWrapper (events.js:316:30)
    at emitOne (events.js:115:13)
    at Socket.emit (events.js:210:7)
    at emitErrorNT (internal/streams/destroy.js:64:8)
    at _combinedTickCallback (internal/process/next_tick.js:138:11)
    at process._tickCallback (internal/process/next_tick.js:180:9)
name: 'MongoError'
message: '[egg-mongoose]failed to connect to server [localhost:27017] on first connect [MongoError: connect ECONNREFUSED 127.0.0.1:27017]'
pid: 2230
hostname: Mac-mini.local

2017-10-09 11:36:21,334 ERROR 4502 [-/::1/-/16ms GET /api/files/sdasd] nodejs.ReferenceError: path is not defined
    at FilesController.show (/Users/caigehui/Documents/ezz-example/app/controller/base/files.js:6:25)
    at show.next (<anonymous>)
    at onFulfilled (/Users/caigehui/Documents/ezz-example/node_modules/co/index.js:65:19)
    at /Users/caigehui/Documents/ezz-example/node_modules/co/index.js:54:5
    at Promise (<anonymous>)
    at Object.co (/Users/caigehui/Documents/ezz-example/node_modules/co/index.js:50:10)
    at Object.toPromise (/Users/caigehui/Documents/ezz-example/node_modules/co/index.js:118:63)
    at next (/Users/caigehui/Documents/ezz-example/node_modules/co/index.js:99:29)
    at onFulfilled (/Users/caigehui/Documents/ezz-example/node_modules/co/index.js:69:7)
    at /Users/caigehui/Documents/ezz-example/node_modules/co/index.js:54:5
    at Promise (<anonymous>)
    at Object.co (/Users/caigehui/Documents/ezz-example/node_modules/co/index.js:50:10)
    at Object.toPromise (/Users/caigehui/Documents/ezz-example/node_modules/co/index.js:118:63)
    at next (/Users/caigehui/Documents/ezz-example/node_modules/co/index.js:99:29)
    at onFulfilled (/Users/caigehui/Documents/ezz-example/node_modules/co/index.js:69:7)
    at /Users/caigehui/Documents/ezz-example/node_modules/co/index.js:54:5

pid: 4502
hostname: Mac-mini.local

2017-10-09 11:36:48,063 ERROR 4535 [-/::1/-/13ms GET /api/files/sdasd] nodejs.ReferenceError: fs is not defined
    at FilesController.show (/Users/caigehui/Documents/ezz-example/app/controller/base/files.js:9:19)
    at show.next (<anonymous>)
    at onFulfilled (/Users/caigehui/Documents/ezz-example/node_modules/co/index.js:65:19)
    at /Users/caigehui/Documents/ezz-example/node_modules/co/index.js:54:5
    at Promise (<anonymous>)
    at Object.co (/Users/caigehui/Documents/ezz-example/node_modules/co/index.js:50:10)
    at Object.toPromise (/Users/caigehui/Documents/ezz-example/node_modules/co/index.js:118:63)
    at next (/Users/caigehui/Documents/ezz-example/node_modules/co/index.js:99:29)
    at onFulfilled (/Users/caigehui/Documents/ezz-example/node_modules/co/index.js:69:7)
    at /Users/caigehui/Documents/ezz-example/node_modules/co/index.js:54:5
    at Promise (<anonymous>)
    at Object.co (/Users/caigehui/Documents/ezz-example/node_modules/co/index.js:50:10)
    at Object.toPromise (/Users/caigehui/Documents/ezz-example/node_modules/co/index.js:118:63)
    at next (/Users/caigehui/Documents/ezz-example/node_modules/co/index.js:99:29)
    at onFulfilled (/Users/caigehui/Documents/ezz-example/node_modules/co/index.js:69:7)
    at /Users/caigehui/Documents/ezz-example/node_modules/co/index.js:54:5

pid: 4535
hostname: Mac-mini.local

2017-10-09 11:36:48,180 ERROR 4535 [-/::1/-/2ms GET /api/files/sdasd] nodejs.ReferenceError: fs is not defined
    at FilesController.show (/Users/caigehui/Documents/ezz-example/app/controller/base/files.js:9:19)
    at show.next (<anonymous>)
    at onFulfilled (/Users/caigehui/Documents/ezz-example/node_modules/co/index.js:65:19)
    at /Users/caigehui/Documents/ezz-example/node_modules/co/index.js:54:5
    at Promise (<anonymous>)
    at Object.co (/Users/caigehui/Documents/ezz-example/node_modules/co/index.js:50:10)
    at Object.toPromise (/Users/caigehui/Documents/ezz-example/node_modules/co/index.js:118:63)
    at next (/Users/caigehui/Documents/ezz-example/node_modules/co/index.js:99:29)
    at onFulfilled (/Users/caigehui/Documents/ezz-example/node_modules/co/index.js:69:7)
    at /Users/caigehui/Documents/ezz-example/node_modules/co/index.js:54:5
    at Promise (<anonymous>)
    at Object.co (/Users/caigehui/Documents/ezz-example/node_modules/co/index.js:50:10)
    at Object.toPromise (/Users/caigehui/Documents/ezz-example/node_modules/co/index.js:118:63)
    at next (/Users/caigehui/Documents/ezz-example/node_modules/co/index.js:99:29)
    at onFulfilled (/Users/caigehui/Documents/ezz-example/node_modules/co/index.js:69:7)
    at /Users/caigehui/Documents/ezz-example/node_modules/co/index.js:54:5

pid: 4535
hostname: Mac-mini.local

2017-10-09 11:36:53,198 ERROR 4535 [-/::1/-/2ms GET /api/files/sdasd] nodejs.ReferenceError: fs is not defined
    at FilesController.show (/Users/caigehui/Documents/ezz-example/app/controller/base/files.js:9:19)
    at show.next (<anonymous>)
    at onFulfilled (/Users/caigehui/Documents/ezz-example/node_modules/co/index.js:65:19)
    at /Users/caigehui/Documents/ezz-example/node_modules/co/index.js:54:5
    at Promise (<anonymous>)
    at Object.co (/Users/caigehui/Documents/ezz-example/node_modules/co/index.js:50:10)
    at Object.toPromise (/Users/caigehui/Documents/ezz-example/node_modules/co/index.js:118:63)
    at next (/Users/caigehui/Documents/ezz-example/node_modules/co/index.js:99:29)
    at onFulfilled (/Users/caigehui/Documents/ezz-example/node_modules/co/index.js:69:7)
    at /Users/caigehui/Documents/ezz-example/node_modules/co/index.js:54:5
    at Promise (<anonymous>)
    at Object.co (/Users/caigehui/Documents/ezz-example/node_modules/co/index.js:50:10)
    at Object.toPromise (/Users/caigehui/Documents/ezz-example/node_modules/co/index.js:118:63)
    at next (/Users/caigehui/Documents/ezz-example/node_modules/co/index.js:99:29)
    at onFulfilled (/Users/caigehui/Documents/ezz-example/node_modules/co/index.js:69:7)
    at /Users/caigehui/Documents/ezz-example/node_modules/co/index.js:54:5

pid: 4535
hostname: Mac-mini.local

2017-10-09 12:38:59,018 ERROR 5633 [-/127.0.0.1/-/58ms POST /api/files] nodejs.TypeError: Cannot read property 'File' of undefined
    at File.createFile (/Users/caigehui/Documents/ezz-example/app/service/base/file.js:7:42)
    at createFile.next (<anonymous>)
    at onFulfilled (/Users/caigehui/Documents/ezz-example/node_modules/co/index.js:65:19)
    at <anonymous>
    at process._tickCallback (internal/process/next_tick.js:188:7)

pid: 5633
hostname: Mac-mini.local

2017-10-09 12:39:21,865 ERROR 5663 [-/127.0.0.1/-/71ms POST /api/files] nodejs.ValidationError: File validation failed: filename: Cast to Number failed for value "background.jpg" at path "filename", filesize: Path `filesize` is required.
    at ValidationError (/Users/caigehui/Documents/ezz-example/node_modules/mongoose/lib/error/validation.js:28:11)
    at model.Document.invalidate (/Users/caigehui/Documents/ezz-example/node_modules/mongoose/lib/document.js:1602:32)
    at model.Document.set (/Users/caigehui/Documents/ezz-example/node_modules/mongoose/lib/document.js:765:10)
    at model._handleIndex (/Users/caigehui/Documents/ezz-example/node_modules/mongoose/lib/document.js:598:14)
    at model.Document.set (/Users/caigehui/Documents/ezz-example/node_modules/mongoose/lib/document.js:558:24)
    at model.Document (/Users/caigehui/Documents/ezz-example/node_modules/mongoose/lib/document.js:76:12)
    at model.Model (/Users/caigehui/Documents/ezz-example/node_modules/mongoose/lib/model.js:47:12)
    at new model (/Users/caigehui/Documents/ezz-example/node_modules/mongoose/lib/model.js:3701:13)
    at /Users/caigehui/Documents/ezz-example/node_modules/mongoose/lib/model.js:1945:51
    at /Users/caigehui/Documents/ezz-example/node_modules/mongoose/node_modules/async/internal/parallel.js:27:9
    at eachOfArrayLike (/Users/caigehui/Documents/ezz-example/node_modules/mongoose/node_modules/async/eachOf.js:57:9)
    at exports.default (/Users/caigehui/Documents/ezz-example/node_modules/mongoose/node_modules/async/eachOf.js:9:5)
    at _parallel (/Users/caigehui/Documents/ezz-example/node_modules/mongoose/node_modules/async/internal/parallel.js:26:5)
    at parallelLimit (/Users/caigehui/Documents/ezz-example/node_modules/mongoose/node_modules/async/parallel.js:85:26)
    at /Users/caigehui/Documents/ezz-example/node_modules/mongoose/lib/model.js:1966:5
    at Promise (<anonymous>)
errors: { filename:  { CastError: Cast to Number failed for value "background.jpg" at path "filename" at CastError (/Users/caigehui/Documents/ezz-example/node_modules/mongoose/lib/error/cast.js:27:11) at model.Document.set (/Users/caigehui/Documents/ezz-example/node_modules/mongoose/lib/document.js:766:7) at model._handleIndex (/Users/caigehui/Documents/ezz-example/node_modules/mongoose/lib/document.js:598:14) at model.Document.set (/Users/caigehui/Documents/ezz-example/node_modules/mongoose/lib/document.js:558:24) at model.Document (/Users/caigehui/Documents/ezz-example/node_modules/mongoose/lib/document.js:76:12) at model.Model (/Users/caigehui/Documents/ezz-example/node_modules/mongoose/lib/model.js:47:12) at new model (/Users/caigehui/Documents/ezz-example/node_modules/mongoose/lib/model.js:3701:13) at /Users/caigehui/Documents/ezz-example/node_modules/mongoose/lib/model.js:1945:51 at /Users/caigehui/Documents/ezz-example/node_modules/mongoose/node_modules/async/internal/parallel.js:27:9 at eachOfArrayLike (/Users/caigehui/Documents/ezz-example/node_modules/mongoose/node_modules/async/eachOf.js:57:9) at exports.default (/Users/caigehui/Documents/ezz-example/node_modules/mongoose/node_modules/async/eachOf.js:9:5) at _parallel (/Users/caigehui/Documents/ezz-example/node_modules/mongoose/node_modules/async/internal/parallel.js:26:5) at parallelLimit (/Users/caigehui/Documents/ezz-example/node_modules/mongoose/node_modules/async/parallel.js:85:26) at /Users/caigehui/Documents/ezz-example/node_modules/mongoose/lib/model.js:1966:5 at Promise (<anonymous>) at Function.create (/Users/caigehui/Documents/ezz-example/node_modules/mongoose/lib/model.js:1929:17) message: 'Cast to Number failed for value "background.jpg" at path "filename"', name: 'CastError', stringValue: '"background.jpg"', kind: 'Number', value: 'background.jpg', path: 'filename', reason:  { CastError: Cast to number failed for value "background.jpg" at path "filename" at CastError (/Users/caigehui/Documents/ezz-example/node_modules/mongoose/lib/error/cast.js:27:11) at SchemaNumber.cast (/Users/caigehui/Documents/ezz-example/node_modules/mongoose/lib/schema/number.js:231:9) at SchemaNumber.SchemaType.applySetters (/Users/caigehui/Documents/ezz-example/node_modules/mongoose/lib/schematype.js:681:12) at model.Document.set (/Users/caigehui/Documents/ezz-example/node_modules/mongoose/lib/document.js:757:18) at model._handleIndex (/Users/caigehui/Documents/ezz-example/node_modules/mongoose/lib/document.js:598:14) at model.Document.set (/Users/caigehui/Documents/ezz-example/node_modules/mongoose/lib/document.js:558:24) at model.Document (/Users/caigehui/Documents/ezz-example/node_modules/mongoose/lib/document.js:76:12) at model.Model (/Users/caigehui/Documents/ezz-example/node_modules/mongoose/lib/model.js:47:12) at new model (/Users/caigehui/Documents/ezz-example/node_modules/mongoose/lib/model.js:3701:13) at /Users/caigehui/Documents/ezz-example/node_modules/mongoose/lib/model.js:1945:51 at /Users/caigehui/Documents/ezz-example/node_modules/mongoose/node_modules/async/internal/parallel.js:27:9 at eachOfArrayLike (/Users/caigehui/Documents/ezz-example/node_modules/mongoose/node_modules/async/eachOf.js:57:9) at exports.default (/Users/caigehui/Documents/ezz-example/node_modules/mongoose/node_modules/async/eachOf.js:9:5) at _parallel (/Users/caigehui/Documents/ezz-example/node_modules/mongoose/node_modules/async/internal/parallel.js:26:5) at parallelLimit (/Users/caigehui/Documents/ezz-example/node_modules/mongoose/node_modules/async/parallel.js:85:26) at /Users/caigehui/Documents/ezz-example/node_modules/mongoose/lib/model.js:1966:5 message: 'Cast to number failed for value "background.jpg" at path "filename"', name: 'CastError', stringValue: '"background.jpg"', kind: 'number', value: 'background.jpg', path: 'filename', reason: undefined } }, filesize:  { ValidatorError: Path `filesize` is required. at ValidatorError (/Users/caigehui/Documents/ezz-example/node_modules/mongoose/lib/error/validator.js:25:11) at validate (/Users/caigehui/Documents/ezz-example/node_modules/mongoose/lib/schematype.js:758:13) at /Users/caigehui/Documents/ezz-example/node_modules/mongoose/lib/schematype.js:805:11 at Array.forEach (<anonymous>) at SchemaNumber.SchemaType.doValidate (/Users/caigehui/Documents/ezz-example/node_modules/mongoose/lib/schematype.js:765:19) at /Users/caigehui/Documents/ezz-example/node_modules/mongoose/lib/document.js:1472:9 at _combinedTickCallback (internal/process/next_tick.js:131:7) at process._tickCallback (internal/process/next_tick.js:180:9) message: 'Path `filesize` is required.', name: 'ValidatorError', properties:  { type: 'required', message: 'Path `{PATH}` is required.', validator: [Function], path: 'filesize', value: undefined }, kind: 'required', path: 'filesize', value: undefined, reason: undefined, '$isValidatorError': true } }
_message: 'File validation failed'
message: 'File validation failed: filename: Cast to Number failed for value "background.jpg" at path "filename", filesize: Path `filesize` is required.'
name: 'ValidationError'
pid: 5663
hostname: Mac-mini.local

2017-10-09 12:41:57,861 ERROR 5726 [-/127.0.0.1/-/69ms POST /api/files] nodejs.ValidationError: File validation failed: filesize: Path `filesize` is required.
    at ValidationError (/Users/caigehui/Documents/ezz-example/node_modules/mongoose/lib/error/validation.js:28:11)
    at model.Document.invalidate (/Users/caigehui/Documents/ezz-example/node_modules/mongoose/lib/document.js:1602:32)
    at /Users/caigehui/Documents/ezz-example/node_modules/mongoose/lib/document.js:1474:17
    at validate (/Users/caigehui/Documents/ezz-example/node_modules/mongoose/lib/schematype.js:760:7)
    at /Users/caigehui/Documents/ezz-example/node_modules/mongoose/lib/schematype.js:805:11
    at Array.forEach (<anonymous>)
    at SchemaNumber.SchemaType.doValidate (/Users/caigehui/Documents/ezz-example/node_modules/mongoose/lib/schematype.js:765:19)
    at /Users/caigehui/Documents/ezz-example/node_modules/mongoose/lib/document.js:1472:9
    at _combinedTickCallback (internal/process/next_tick.js:131:7)
    at process._tickCallback (internal/process/next_tick.js:180:9)
errors: { filesize:  { ValidatorError: Path `filesize` is required. at ValidatorError (/Users/caigehui/Documents/ezz-example/node_modules/mongoose/lib/error/validator.js:25:11) at validate (/Users/caigehui/Documents/ezz-example/node_modules/mongoose/lib/schematype.js:758:13) at /Users/caigehui/Documents/ezz-example/node_modules/mongoose/lib/schematype.js:805:11 at Array.forEach (<anonymous>) at SchemaNumber.SchemaType.doValidate (/Users/caigehui/Documents/ezz-example/node_modules/mongoose/lib/schematype.js:765:19) at /Users/caigehui/Documents/ezz-example/node_modules/mongoose/lib/document.js:1472:9 at _combinedTickCallback (internal/process/next_tick.js:131:7) at process._tickCallback (internal/process/next_tick.js:180:9) message: 'Path `filesize` is required.', name: 'ValidatorError', properties:  { type: 'required', message: 'Path `{PATH}` is required.', validator: [Function], path: 'filesize', value: undefined }, kind: 'required', path: 'filesize', value: undefined, reason: undefined, '$isValidatorError': true } }
_message: 'File validation failed'
message: 'File validation failed: filesize: Path `filesize` is required.'
name: 'ValidationError'
pid: 5726
hostname: Mac-mini.local

2017-10-09 12:42:56,029 ERROR 5793 [-/127.0.0.1/-/65ms POST /api/files] nodejs.ValidationError: File validation failed: filesize: Path `filesize` is required.
    at ValidationError (/Users/caigehui/Documents/ezz-example/node_modules/mongoose/lib/error/validation.js:28:11)
    at model.Document.invalidate (/Users/caigehui/Documents/ezz-example/node_modules/mongoose/lib/document.js:1602:32)
    at /Users/caigehui/Documents/ezz-example/node_modules/mongoose/lib/document.js:1474:17
    at validate (/Users/caigehui/Documents/ezz-example/node_modules/mongoose/lib/schematype.js:760:7)
    at /Users/caigehui/Documents/ezz-example/node_modules/mongoose/lib/schematype.js:805:11
    at Array.forEach (<anonymous>)
    at SchemaNumber.SchemaType.doValidate (/Users/caigehui/Documents/ezz-example/node_modules/mongoose/lib/schematype.js:765:19)
    at /Users/caigehui/Documents/ezz-example/node_modules/mongoose/lib/document.js:1472:9
    at _combinedTickCallback (internal/process/next_tick.js:131:7)
    at process._tickCallback (internal/process/next_tick.js:180:9)
errors: { filesize:  { ValidatorError: Path `filesize` is required. at ValidatorError (/Users/caigehui/Documents/ezz-example/node_modules/mongoose/lib/error/validator.js:25:11) at validate (/Users/caigehui/Documents/ezz-example/node_modules/mongoose/lib/schematype.js:758:13) at /Users/caigehui/Documents/ezz-example/node_modules/mongoose/lib/schematype.js:805:11 at Array.forEach (<anonymous>) at SchemaNumber.SchemaType.doValidate (/Users/caigehui/Documents/ezz-example/node_modules/mongoose/lib/schematype.js:765:19) at /Users/caigehui/Documents/ezz-example/node_modules/mongoose/lib/document.js:1472:9 at _combinedTickCallback (internal/process/next_tick.js:131:7) at process._tickCallback (internal/process/next_tick.js:180:9) message: 'Path `filesize` is required.', name: 'ValidatorError', properties:  { type: 'required', message: 'Path `{PATH}` is required.', validator: [Function], path: 'filesize', value: undefined }, kind: 'required', path: 'filesize', value: undefined, reason: undefined, '$isValidatorError': true } }
_message: 'File validation failed'
message: 'File validation failed: filesize: Path `filesize` is required.'
name: 'ValidationError'
pid: 5793
hostname: Mac-mini.local

2017-10-09 12:43:25,543 ERROR 5825 [-/127.0.0.1/-/270ms POST /api/files] nodejs.ValidationError: File validation failed: filesize: Path `filesize` is required.
    at ValidationError (/Users/caigehui/Documents/ezz-example/node_modules/mongoose/lib/error/validation.js:28:11)
    at model.Document.invalidate (/Users/caigehui/Documents/ezz-example/node_modules/mongoose/lib/document.js:1602:32)
    at /Users/caigehui/Documents/ezz-example/node_modules/mongoose/lib/document.js:1474:17
    at validate (/Users/caigehui/Documents/ezz-example/node_modules/mongoose/lib/schematype.js:760:7)
    at /Users/caigehui/Documents/ezz-example/node_modules/mongoose/lib/schematype.js:805:11
    at Array.forEach (<anonymous>)
    at SchemaNumber.SchemaType.doValidate (/Users/caigehui/Documents/ezz-example/node_modules/mongoose/lib/schematype.js:765:19)
    at /Users/caigehui/Documents/ezz-example/node_modules/mongoose/lib/document.js:1472:9
    at _combinedTickCallback (internal/process/next_tick.js:131:7)
    at process._tickCallback (internal/process/next_tick.js:180:9)
errors: { filesize:  { ValidatorError: Path `filesize` is required. at ValidatorError (/Users/caigehui/Documents/ezz-example/node_modules/mongoose/lib/error/validator.js:25:11) at validate (/Users/caigehui/Documents/ezz-example/node_modules/mongoose/lib/schematype.js:758:13) at /Users/caigehui/Documents/ezz-example/node_modules/mongoose/lib/schematype.js:805:11 at Array.forEach (<anonymous>) at SchemaNumber.SchemaType.doValidate (/Users/caigehui/Documents/ezz-example/node_modules/mongoose/lib/schematype.js:765:19) at /Users/caigehui/Documents/ezz-example/node_modules/mongoose/lib/document.js:1472:9 at _combinedTickCallback (internal/process/next_tick.js:131:7) at process._tickCallback (internal/process/next_tick.js:180:9) message: 'Path `filesize` is required.', name: 'ValidatorError', properties:  { type: 'required', message: 'Path `{PATH}` is required.', validator: [Function], path: 'filesize', value: undefined }, kind: 'required', path: 'filesize', value: undefined, reason: undefined, '$isValidatorError': true } }
_message: 'File validation failed'
message: 'File validation failed: filesize: Path `filesize` is required.'
name: 'ValidationError'
pid: 5825
hostname: Mac-mini.local

2017-10-09 13:00:03,700 ERROR 6280 [-/127.0.0.1/-/42ms POST /api/files] nodejs.TypeError: this.ctx.helper.getFileSize is not a function
    at File.createFile (/Users/caigehui/Documents/ezz-example/app/service/base/file.js:7:52)
    at createFile.next (<anonymous>)
    at onFulfilled (/Users/caigehui/Documents/ezz-example/node_modules/co/index.js:65:19)
    at <anonymous>
    at process._tickCallback (internal/process/next_tick.js:188:7)

pid: 6280
hostname: Mac-mini.local

2017-10-09 13:00:26,127 ERROR 6318 [-/127.0.0.1/-/175ms POST /api/files] nodejs.ValidationError: File validation failed: filesize: Path `filesize` is required.
    at ValidationError (/Users/caigehui/Documents/ezz-example/node_modules/mongoose/lib/error/validation.js:28:11)
    at model.Document.invalidate (/Users/caigehui/Documents/ezz-example/node_modules/mongoose/lib/document.js:1602:32)
    at /Users/caigehui/Documents/ezz-example/node_modules/mongoose/lib/document.js:1474:17
    at validate (/Users/caigehui/Documents/ezz-example/node_modules/mongoose/lib/schematype.js:760:7)
    at /Users/caigehui/Documents/ezz-example/node_modules/mongoose/lib/schematype.js:805:11
    at Array.forEach (<anonymous>)
    at SchemaNumber.SchemaType.doValidate (/Users/caigehui/Documents/ezz-example/node_modules/mongoose/lib/schematype.js:765:19)
    at /Users/caigehui/Documents/ezz-example/node_modules/mongoose/lib/document.js:1472:9
    at _combinedTickCallback (internal/process/next_tick.js:131:7)
    at process._tickCallback (internal/process/next_tick.js:180:9)
errors: { filesize:  { ValidatorError: Path `filesize` is required. at ValidatorError (/Users/caigehui/Documents/ezz-example/node_modules/mongoose/lib/error/validator.js:25:11) at validate (/Users/caigehui/Documents/ezz-example/node_modules/mongoose/lib/schematype.js:758:13) at /Users/caigehui/Documents/ezz-example/node_modules/mongoose/lib/schematype.js:805:11 at Array.forEach (<anonymous>) at SchemaNumber.SchemaType.doValidate (/Users/caigehui/Documents/ezz-example/node_modules/mongoose/lib/schematype.js:765:19) at /Users/caigehui/Documents/ezz-example/node_modules/mongoose/lib/document.js:1472:9 at _combinedTickCallback (internal/process/next_tick.js:131:7) at process._tickCallback (internal/process/next_tick.js:180:9) message: 'Path `filesize` is required.', name: 'ValidatorError', properties:  { type: 'required', message: 'Path `{PATH}` is required.', validator: [Function], path: 'filesize', value: undefined }, kind: 'required', path: 'filesize', value: undefined, reason: undefined, '$isValidatorError': true } }
_message: 'File validation failed'
message: 'File validation failed: filesize: Path `filesize` is required.'
name: 'ValidationError'
pid: 6318
hostname: Mac-mini.local

2017-10-09 14:12:19,424 ERROR 6926 [-/127.0.0.1/-/15ms GET /api/files/ds-s-s] nodejs.TypeError: Path must be a string. Received undefined
    at assertPath (path.js:28:11)
    at Object.join (path.js:1249:7)
    at Object.config.getUploadPath.filename [as getUploadPath] (/Users/caigehui/Documents/ezz-example/config/config.default.js:49:21)
    at FilesController.show (/Users/caigehui/Documents/ezz-example/app/controller/base/files.js:13:32)
    at show.next (<anonymous>)
    at onFulfilled (/Users/caigehui/Documents/ezz-example/node_modules/co/index.js:65:19)
    at /Users/caigehui/Documents/ezz-example/node_modules/co/index.js:54:5
    at Promise (<anonymous>)
    at Object.co (/Users/caigehui/Documents/ezz-example/node_modules/co/index.js:50:10)
    at Object.toPromise (/Users/caigehui/Documents/ezz-example/node_modules/co/index.js:118:63)
    at next (/Users/caigehui/Documents/ezz-example/node_modules/co/index.js:99:29)
    at onFulfilled (/Users/caigehui/Documents/ezz-example/node_modules/co/index.js:69:7)
    at /Users/caigehui/Documents/ezz-example/node_modules/co/index.js:54:5
    at Promise (<anonymous>)
    at Object.co (/Users/caigehui/Documents/ezz-example/node_modules/co/index.js:50:10)
    at Object.toPromise (/Users/caigehui/Documents/ezz-example/node_modules/co/index.js:118:63)

pid: 6926
hostname: Mac-mini.local

2017-10-09 14:13:51,222 ERROR 7009 [-/127.0.0.1/-/30ms GET /api/files/ds-s-s] nodejs.TypeError: Path must be a string. Received undefined
    at assertPath (path.js:28:11)
    at Object.join (path.js:1249:7)
    at Object.config.getUploadPath.filename [as getUploadPath] (/Users/caigehui/Documents/ezz-example/config/config.default.js:49:21)
    at FilesController.show (/Users/caigehui/Documents/ezz-example/app/controller/base/files.js:14:32)
    at show.next (<anonymous>)
    at onFulfilled (/Users/caigehui/Documents/ezz-example/node_modules/co/index.js:65:19)
    at /Users/caigehui/Documents/ezz-example/node_modules/co/index.js:54:5
    at Promise (<anonymous>)
    at Object.co (/Users/caigehui/Documents/ezz-example/node_modules/co/index.js:50:10)
    at Object.toPromise (/Users/caigehui/Documents/ezz-example/node_modules/co/index.js:118:63)
    at next (/Users/caigehui/Documents/ezz-example/node_modules/co/index.js:99:29)
    at onFulfilled (/Users/caigehui/Documents/ezz-example/node_modules/co/index.js:69:7)
    at /Users/caigehui/Documents/ezz-example/node_modules/co/index.js:54:5
    at Promise (<anonymous>)
    at Object.co (/Users/caigehui/Documents/ezz-example/node_modules/co/index.js:50:10)
    at Object.toPromise (/Users/caigehui/Documents/ezz-example/node_modules/co/index.js:118:63)

pid: 7009
hostname: Mac-mini.local

2017-10-09 14:14:16,418 ERROR 7031 [-/127.0.0.1/-/21ms GET /api/files/ds-s-s] nodejs.CastError: Cast to ObjectId failed for value "ds-s-s" at path "_id" for model "File"
    at CastError (/Users/caigehui/Documents/ezz-example/node_modules/mongoose/lib/error/cast.js:27:11)
    at ObjectId.cast (/Users/caigehui/Documents/ezz-example/node_modules/mongoose/lib/schema/objectid.js:158:13)
    at ObjectId.SchemaType._castForQuery (/Users/caigehui/Documents/ezz-example/node_modules/mongoose/lib/schematype.js:1064:15)
    at ObjectId.castForQuery (/Users/caigehui/Documents/ezz-example/node_modules/mongoose/lib/schema/objectid.js:198:15)
    at ObjectId.SchemaType.castForQueryWrapper (/Users/caigehui/Documents/ezz-example/node_modules/mongoose/lib/schematype.js:1021:15)
    at cast (/Users/caigehui/Documents/ezz-example/node_modules/mongoose/lib/cast.js:269:32)
    at model.Query.Query.cast (/Users/caigehui/Documents/ezz-example/node_modules/mongoose/lib/query.js:3109:12)
    at model.Query.Query._castConditions (/Users/caigehui/Documents/ezz-example/node_modules/mongoose/lib/query.js:1145:10)
    at model.Query.Query._findOne (/Users/caigehui/Documents/ezz-example/node_modules/mongoose/lib/query.js:1352:8)
    at /Users/caigehui/Documents/ezz-example/node_modules/kareem/index.js:250:8
    at /Users/caigehui/Documents/ezz-example/node_modules/kareem/index.js:23:7
    at _combinedTickCallback (internal/process/next_tick.js:131:7)
    at process._tickCallback (internal/process/next_tick.js:180:9)
message: 'Cast to ObjectId failed for value "ds-s-s" at path "_id" for model "File"'
name: 'CastError'
stringValue: '"ds-s-s"'
kind: 'ObjectId'
value: 'ds-s-s'
path: '_id'
reason: undefined
model: { [Function: model]
  hooks: Kareem { _pres: {}, _posts: {} },
  base: 
   Mongoose {
     connections: [ [Object], [Object] ],
     models: {},
     modelSchemas: { File: [Object], User: [Object] },
     options: { pluralization: true },
     plugins: [ [Array], [Array], [Array] ] },
  modelName: 'File',
  model: [Function: model],
  db: 
   NativeConnection {
     base: 
      Mongoose {
        connections: [Array],
        models: {},
        modelSchemas: [Object],
        options: [Object],
        plugins: [Array] },
     collections: { files: [Object], users: [Object] },
     models: { File: [Object], User: [Object] },
     config: { autoIndex: true },
     replica: false,
     hosts: null,
     host: 'localhost',
     port: 27017,
     user: undefined,
     pass: undefined,
     name: 'ezz',
     options: 
      { db: [Object],
        auth: {},
        server: [Object],
        replset: [Object],
        mongos: undefined },
     otherDbs: [],
     states: 
      { '0': 'disconnected',
        '1': 'connected',
        '2': 'connecting',
        '3': 'disconnecting',
        '4': 'unauthorized',
        '99': 'uninitialized',
        disconnected: 0,
        connected: 1,
        connecting: 2,
        disconnecting: 3,
        unauthorized: 4,
        uninitialized: 99 },
     _readyState: 1,
     _closeCalled: false,
     _hasOpened: true,
     _listening: false,
     db: 
      Db {
        domain: null,
        _events: [Object],
        _eventsCount: 6,
        _maxListeners: undefined,
        s: [Object],
        serverConfig: [Getter],
        bufferMaxEntries: [Getter],
        databaseName: [Getter],
        _listening: true },
     Schema: 
      { [Function: Schema]
        reserved: [Object],
        interpretAsType: [Function],
        Types: [Object],
        ObjectId: [Object] },
     _events: 
      { error: [Function],
        disconnected: [Function],
        connected: [Function],
        reconnected: [Function] },
     _eventsCount: 4 },
  discriminators: undefined,
  '$appliedHooks': true,
  _events: { init: [Function], save: [Function] },
  _eventsCount: 2,
  schema: 
   Schema {
     obj: 
      { filename: [Object],
        filesize: [Object],
        hash: [Object],
        temp: [Object] },
     paths: 
      { filename: [Object],
        filesize: [Object],
        hash: [Object],
        temp: [Object],
        _id: [Object],
        __v: [Object] },
     aliases: {},
     subpaths: {},
     virtuals: { id: [Object] },
     singleNestedPaths: {},
     nested: {},
     inherits: {},
     callQueue: [ [Array], [Array], [Array], [Array], [Array], [Array] ],
     _indexes: [],
     methods: {},
     statics: {},
     tree: 
      { filename: [Object],
        filesize: [Object],
        hash: [Object],
        temp: [Object],
        _id: [Object],
        __v: [Function: Number],
        id: [Object] },
     query: {},
     childSchemas: [],
     plugins: [ [Object], [Object], [Object], [Object] ],
     s: { hooks: [Object], kareemHooks: [Object] },
     options: 
      { retainKeyOrder: false,
        typeKey: 'type',
        id: true,
        noVirtualId: false,
        _id: true,
        noId: false,
        validateBeforeSave: true,
        read: null,
        shardKey: null,
        autoIndex: null,
        minimize: true,
        discriminatorKey: '__t',
        versionKey: '__v',
        capped: false,
        bufferCommands: true,
        strict: true,
        pluralization: true },
     '$globalPluginsApplied': true },
  collection: 
   NativeCollection {
     collection: Collection { s: [Object] },
     opts: { bufferCommands: true, capped: false },
     name: 'files',
     collectionName: 'files',
     conn: 
      NativeConnection {
        base: [Object],
        collections: [Object],
        models: [Object],
        config: [Object],
        replica: false,
        hosts: null,
        host: 'localhost',
        port: 27017,
        user: undefined,
        pass: undefined,
        name: 'ezz',
        options: [Object],
        otherDbs: [],
        states: [Object],
        _readyState: 1,
        _closeCalled: false,
        _hasOpened: true,
        _listening: false,
        db: [Object],
        Schema: [Object],
        _events: [Object],
        _eventsCount: 4 },
     queue: [],
     buffer: false,
     emitter: 
      EventEmitter {
        domain: null,
        _events: {},
        _eventsCount: 0,
        _maxListeners: undefined } },
  Query: 
   { [Function]
     base: 
      Query {
        toConstructor: [Function: toConstructor],
        setOptions: [Function],
        collection: [Function: collection],
        '$where': [Function],
        where: [Function],
        equals: [Function: equals],
        eq: [Function: eq],
        or: [Function: or],
        nor: [Function: nor],
        and: [Function: and],
        gt: [Function],
        gte: [Function],
        lt: [Function],
        lte: [Function],
        ne: [Function],
        in: [Function],
        nin: [Function],
        all: [Function],
        regex: [Function],
        size: [Function],
        maxDistance: [Function],
        minDistance: [Function],
        mod: [Function],
        exists: [Function],
        elemMatch: [Function],
        within: [Function: within],
        box: [Function],
        polygon: [Function],
        circle: [Function],
        near: [Function: near],
        intersects: [Function: intersects],
        geometry: [Function: geometry],
        select: [Function: select],
        slice: [Function],
        sort: [Function],
        limit: [Function],
        skip: [Function],
        maxScan: [Function],
        batchSize: [Function],
        comment: [Function],
        maxTime: [Function],
        snapshot: [Function],
        hint: [Function],
        slaveOk: [Function],
        read: [Function],
        tailable: [Function],
        merge: [Function],
        find: [Function],
        cursor: [Function: cursor],
        findOne: [Function],
        count: [Function],
        distinct: [Function],
        update: [Function: update],
        updateMany: [Function: updateMany],
        updateOne: [Function: updateOne],
        replaceOne: [Function: replaceOne],
        remove: [Function],
        deleteOne: [Function],
        deleteMany: [Function],
        findOneAndUpdate: [Function],
        findOneAndRemove: [Function],
        _findAndModify: [Function],
        _wrapCallback: [Function],
        setTraceFunction: [Function],
        exec: [Function: exec],
        thunk: [Function],
        then: [Function],
        stream: [Function],
        selected: [Function: selected],
        selectedInclusively: [Function: selectedInclusively],
        selectedExclusively: [Function: selectedExclusively],
        _mergeUpdate: [Function],
        _optionsForExec: [Function],
        _fieldsForExec: [Function],
        _updateForExec: [Function],
        _ensurePath: [Function],
        _validate: [Function] } },
  '$__insertMany': [Function],
  insertMany: [Function],
  [Symbol(EGG_LOADER_ITEM_FULLPATH)]: '/Users/caigehui/Documents/ezz-example/app/model/file.js',
  [Symbol(EGG_LOADER_ITEM_EXPORTS)]: true }
pid: 7031
hostname: Mac-mini.local

2017-10-09 14:18:37,205 ERROR 7117 [-/127.0.0.1/-/22ms GET /api/files/ds-s-s] nodejs.CastError: Cast to ObjectId failed for value "ds" at path "_id" for model "File"
    at CastError (/Users/caigehui/Documents/ezz-example/node_modules/mongoose/lib/error/cast.js:27:11)
    at ObjectId.cast (/Users/caigehui/Documents/ezz-example/node_modules/mongoose/lib/schema/objectid.js:158:13)
    at ObjectId.SchemaType._castForQuery (/Users/caigehui/Documents/ezz-example/node_modules/mongoose/lib/schematype.js:1064:15)
    at ObjectId.castForQuery (/Users/caigehui/Documents/ezz-example/node_modules/mongoose/lib/schema/objectid.js:198:15)
    at ObjectId.SchemaType.castForQueryWrapper (/Users/caigehui/Documents/ezz-example/node_modules/mongoose/lib/schematype.js:1021:15)
    at cast (/Users/caigehui/Documents/ezz-example/node_modules/mongoose/lib/cast.js:269:32)
    at model.Query.Query.cast (/Users/caigehui/Documents/ezz-example/node_modules/mongoose/lib/query.js:3109:12)
    at model.Query.Query._castConditions (/Users/caigehui/Documents/ezz-example/node_modules/mongoose/lib/query.js:1145:10)
    at model.Query.Query._findOne (/Users/caigehui/Documents/ezz-example/node_modules/mongoose/lib/query.js:1352:8)
    at /Users/caigehui/Documents/ezz-example/node_modules/kareem/index.js:250:8
    at /Users/caigehui/Documents/ezz-example/node_modules/kareem/index.js:23:7
    at _combinedTickCallback (internal/process/next_tick.js:131:7)
    at process._tickCallback (internal/process/next_tick.js:180:9)
message: 'Cast to ObjectId failed for value "ds" at path "_id" for model "File"'
name: 'CastError'
stringValue: '"ds"'
kind: 'ObjectId'
value: 'ds'
path: '_id'
reason: undefined
model: { [Function: model]
  hooks: Kareem { _pres: {}, _posts: {} },
  base: 
   Mongoose {
     connections: [ [Object], [Object] ],
     models: {},
     modelSchemas: { File: [Object], User: [Object] },
     options: { pluralization: true },
     plugins: [ [Array], [Array], [Array] ] },
  modelName: 'File',
  model: [Function: model],
  db: 
   NativeConnection {
     base: 
      Mongoose {
        connections: [Array],
        models: {},
        modelSchemas: [Object],
        options: [Object],
        plugins: [Array] },
     collections: { files: [Object], users: [Object] },
     models: { File: [Object], User: [Object] },
     config: { autoIndex: true },
     replica: false,
     hosts: null,
     host: 'localhost',
     port: 27017,
     user: undefined,
     pass: undefined,
     name: 'ezz',
     options: 
      { db: [Object],
        auth: {},
        server: [Object],
        replset: [Object],
        mongos: undefined },
     otherDbs: [],
     states: 
      { '0': 'disconnected',
        '1': 'connected',
        '2': 'connecting',
        '3': 'disconnecting',
        '4': 'unauthorized',
        '99': 'uninitialized',
        disconnected: 0,
        connected: 1,
        connecting: 2,
        disconnecting: 3,
        unauthorized: 4,
        uninitialized: 99 },
     _readyState: 1,
     _closeCalled: false,
     _hasOpened: true,
     _listening: false,
     db: 
      Db {
        domain: null,
        _events: [Object],
        _eventsCount: 6,
        _maxListeners: undefined,
        s: [Object],
        serverConfig: [Getter],
        bufferMaxEntries: [Getter],
        databaseName: [Getter],
        _listening: true },
     Schema: 
      { [Function: Schema]
        reserved: [Object],
        interpretAsType: [Function],
        Types: [Object],
        ObjectId: [Object] },
     _events: 
      { error: [Function],
        disconnected: [Function],
        connected: [Function],
        reconnected: [Function] },
     _eventsCount: 4 },
  discriminators: undefined,
  '$appliedHooks': true,
  _events: { init: [Function], save: [Function] },
  _eventsCount: 2,
  schema: 
   Schema {
     obj: 
      { filename: [Object],
        filesize: [Object],
        hash: [Object],
        temp: [Object] },
     paths: 
      { filename: [Object],
        filesize: [Object],
        hash: [Object],
        temp: [Object],
        _id: [Object],
        __v: [Object] },
     aliases: {},
     subpaths: {},
     virtuals: { id: [Object] },
     singleNestedPaths: {},
     nested: {},
     inherits: {},
     callQueue: [ [Array], [Array], [Array], [Array], [Array], [Array] ],
     _indexes: [],
     methods: {},
     statics: {},
     tree: 
      { filename: [Object],
        filesize: [Object],
        hash: [Object],
        temp: [Object],
        _id: [Object],
        __v: [Function: Number],
        id: [Object] },
     query: {},
     childSchemas: [],
     plugins: [ [Object], [Object], [Object], [Object] ],
     s: { hooks: [Object], kareemHooks: [Object] },
     options: 
      { retainKeyOrder: false,
        typeKey: 'type',
        id: true,
        noVirtualId: false,
        _id: true,
        noId: false,
        validateBeforeSave: true,
        read: null,
        shardKey: null,
        autoIndex: null,
        minimize: true,
        discriminatorKey: '__t',
        versionKey: '__v',
        capped: false,
        bufferCommands: true,
        strict: true,
        pluralization: true },
     '$globalPluginsApplied': true },
  collection: 
   NativeCollection {
     collection: Collection { s: [Object] },
     opts: { bufferCommands: true, capped: false },
     name: 'files',
     collectionName: 'files',
     conn: 
      NativeConnection {
        base: [Object],
        collections: [Object],
        models: [Object],
        config: [Object],
        replica: false,
        hosts: null,
        host: 'localhost',
        port: 27017,
        user: undefined,
        pass: undefined,
        name: 'ezz',
        options: [Object],
        otherDbs: [],
        states: [Object],
        _readyState: 1,
        _closeCalled: false,
        _hasOpened: true,
        _listening: false,
        db: [Object],
        Schema: [Object],
        _events: [Object],
        _eventsCount: 4 },
     queue: [],
     buffer: false,
     emitter: 
      EventEmitter {
        domain: null,
        _events: {},
        _eventsCount: 0,
        _maxListeners: undefined } },
  Query: 
   { [Function]
     base: 
      Query {
        toConstructor: [Function: toConstructor],
        setOptions: [Function],
        collection: [Function: collection],
        '$where': [Function],
        where: [Function],
        equals: [Function: equals],
        eq: [Function: eq],
        or: [Function: or],
        nor: [Function: nor],
        and: [Function: and],
        gt: [Function],
        gte: [Function],
        lt: [Function],
        lte: [Function],
        ne: [Function],
        in: [Function],
        nin: [Function],
        all: [Function],
        regex: [Function],
        size: [Function],
        maxDistance: [Function],
        minDistance: [Function],
        mod: [Function],
        exists: [Function],
        elemMatch: [Function],
        within: [Function: within],
        box: [Function],
        polygon: [Function],
        circle: [Function],
        near: [Function: near],
        intersects: [Function: intersects],
        geometry: [Function: geometry],
        select: [Function: select],
        slice: [Function],
        sort: [Function],
        limit: [Function],
        skip: [Function],
        maxScan: [Function],
        batchSize: [Function],
        comment: [Function],
        maxTime: [Function],
        snapshot: [Function],
        hint: [Function],
        slaveOk: [Function],
        read: [Function],
        tailable: [Function],
        merge: [Function],
        find: [Function],
        cursor: [Function: cursor],
        findOne: [Function],
        count: [Function],
        distinct: [Function],
        update: [Function: update],
        updateMany: [Function: updateMany],
        updateOne: [Function: updateOne],
        replaceOne: [Function: replaceOne],
        remove: [Function],
        deleteOne: [Function],
        deleteMany: [Function],
        findOneAndUpdate: [Function],
        findOneAndRemove: [Function],
        _findAndModify: [Function],
        _wrapCallback: [Function],
        setTraceFunction: [Function],
        exec: [Function: exec],
        thunk: [Function],
        then: [Function],
        stream: [Function],
        selected: [Function: selected],
        selectedInclusively: [Function: selectedInclusively],
        selectedExclusively: [Function: selectedExclusively],
        _mergeUpdate: [Function],
        _optionsForExec: [Function],
        _fieldsForExec: [Function],
        _updateForExec: [Function],
        _ensurePath: [Function],
        _validate: [Function] } },
  '$__insertMany': [Function],
  insertMany: [Function],
  [Symbol(EGG_LOADER_ITEM_FULLPATH)]: '/Users/caigehui/Documents/ezz-example/app/model/file.js',
  [Symbol(EGG_LOADER_ITEM_EXPORTS)]: true }
pid: 7117
hostname: Mac-mini.local

2017-10-09 14:25:44,562 ERROR 7349 [-/127.0.0.1/-/35ms GET /api/files/59db027d9f334018d47acc38] nodejs.ENOENTError: ENOENT: no such file or directory, open '/Users/caigehui/Documents/upload/e93d50b3d6b6b4880664c66e4cf4a4c5'
ENOENTError: ENOENT: no such file or directory, open '/Users/caigehui/Documents/upload/e93d50b3d6b6b4880664c66e4cf4a4c5'
errno: -2
code: 'ENOENT'
syscall: 'open'
path: '/Users/caigehui/Documents/upload/e93d50b3d6b6b4880664c66e4cf4a4c5'
name: 'ENOENTError'
pid: 7349
hostname: Mac-mini.local

2017-10-09 14:25:44,989 ERROR 7349 [-/127.0.0.1/-/7ms GET /api/files/59db027d9f334018d47acc38] nodejs.ENOENTError: ENOENT: no such file or directory, open '/Users/caigehui/Documents/upload/e93d50b3d6b6b4880664c66e4cf4a4c5'
ENOENTError: ENOENT: no such file or directory, open '/Users/caigehui/Documents/upload/e93d50b3d6b6b4880664c66e4cf4a4c5'
errno: -2
code: 'ENOENT'
syscall: 'open'
path: '/Users/caigehui/Documents/upload/e93d50b3d6b6b4880664c66e4cf4a4c5'
name: 'ENOENTError'
pid: 7349
hostname: Mac-mini.local

2017-10-09 14:25:50,063 ERROR 7349 [-/127.0.0.1/-/6ms GET /api/files/59db027d9f334018d47acc38] nodejs.ENOENTError: ENOENT: no such file or directory, open '/Users/caigehui/Documents/upload/e93d50b3d6b6b4880664c66e4cf4a4c5'
ENOENTError: ENOENT: no such file or directory, open '/Users/caigehui/Documents/upload/e93d50b3d6b6b4880664c66e4cf4a4c5'
errno: -2
code: 'ENOENT'
syscall: 'open'
path: '/Users/caigehui/Documents/upload/e93d50b3d6b6b4880664c66e4cf4a4c5'
name: 'ENOENTError'
pid: 7349
hostname: Mac-mini.local

2017-10-09 14:26:20,111 ERROR 7349 [-/127.0.0.1/-/2ms GET /api/files/59db027d9f334018d47acc38] nodejs.ENOENTError: ENOENT: no such file or directory, open '/Users/caigehui/Documents/upload/e93d50b3d6b6b4880664c66e4cf4a4c5'
ENOENTError: ENOENT: no such file or directory, open '/Users/caigehui/Documents/upload/e93d50b3d6b6b4880664c66e4cf4a4c5'
errno: -2
code: 'ENOENT'
syscall: 'open'
path: '/Users/caigehui/Documents/upload/e93d50b3d6b6b4880664c66e4cf4a4c5'
name: 'ENOENTError'
pid: 7349
hostname: Mac-mini.local

2017-10-09 14:27:14,992 ERROR 7417 [-/127.0.0.1/-/141ms GET /api/files/59db027d9f334018d47acc38] nodejs.ENOENTError: ENOENT: no such file or directory, open '/Users/caigehui/Documents/upload/e93d50b3d6b6b4880664c66e4cf4a4c5'
ENOENTError: ENOENT: no such file or directory, open '/Users/caigehui/Documents/upload/e93d50b3d6b6b4880664c66e4cf4a4c5'
errno: -2
code: 'ENOENT'
syscall: 'open'
path: '/Users/caigehui/Documents/upload/e93d50b3d6b6b4880664c66e4cf4a4c5'
name: 'ENOENTError'
pid: 7417
hostname: Mac-mini.local

2017-10-09 14:27:33,185 ERROR 7446 [-/127.0.0.1/-/32ms GET /api/files/59db027d9f334018d47acc38] nodejs.ENOENTError: ENOENT: no such file or directory, open '/Users/caigehui/Documents/upload/e93d50b3d6b6b4880664c66e4cf4a4c5'
ENOENTError: ENOENT: no such file or directory, open '/Users/caigehui/Documents/upload/e93d50b3d6b6b4880664c66e4cf4a4c5'
errno: -2
code: 'ENOENT'
syscall: 'open'
path: '/Users/caigehui/Documents/upload/e93d50b3d6b6b4880664c66e4cf4a4c5'
name: 'ENOENTError'
pid: 7446
hostname: Mac-mini.local

2017-10-09 15:41:18,726 ERROR 9288 [-/127.0.0.1/-/147ms POST /api/files] nodejs.TypeError: helper.getUploadPath is not a function
    at File.createFile (/Users/caigehui/Documents/ezz-example/app/service/base/file.js:31:56)
    at createFile.next (<anonymous>)
    at onFulfilled (/Users/caigehui/Documents/ezz-example/node_modules/co/index.js:65:19)
    at <anonymous>

pid: 9288
hostname: Mac-mini.local

2017-10-09 15:42:00,728 ERROR 9318 [-/127.0.0.1/-/153ms POST /api/files] nodejs.TypeError: gm().write() expects a filename when writing new files
    at gm.write (/Users/caigehui/Documents/ezz-example/node_modules/gm/lib/command.js:93:23)
    at Promise (/Users/caigehui/Documents/ezz-example/app/extend/helper.js:110:18)
    at Promise (<anonymous>)
    at Helper.createResizedImage (/Users/caigehui/Documents/ezz-example/app/extend/helper.js:106:16)
    at File.createFile (/Users/caigehui/Documents/ezz-example/app/service/base/file.js:31:30)
    at createFile.next (<anonymous>)
    at onFulfilled (/Users/caigehui/Documents/ezz-example/node_modules/co/index.js:65:19)
    at <anonymous>

pid: 9318
hostname: Mac-mini.local

2017-10-09 15:42:00,740 ERROR 9318 nodejs.ENOENTError: ENOENT: no such file or directory, open '/Users/caigehui/Documents/upload/e93d50b3d6b6b4880664c66e4cf4a4c5-small' (uncaughtException throw 1 times on pid:9318)
ENOENTError: ENOENT: no such file or directory, open '/Users/caigehui/Documents/upload/e93d50b3d6b6b4880664c66e4cf4a4c5-small' (uncaughtException throw 1 times on pid:9318)
errno: -2
code: 'ENOENT'
syscall: 'open'
path: '/Users/caigehui/Documents/upload/e93d50b3d6b6b4880664c66e4cf4a4c5-small'
name: 'ENOENTError'
pid: 9318
hostname: Mac-mini.local

2017-10-09 15:43:37,275 ERROR 9420 [-/127.0.0.1/-/154ms POST /api/files] nodejs.TypeError: gm().write() expects a filename when writing new files
    at gm.write (/Users/caigehui/Documents/ezz-example/node_modules/gm/lib/command.js:93:23)
    at Promise (/Users/caigehui/Documents/ezz-example/app/extend/helper.js:110:18)
    at Promise (<anonymous>)
    at Helper.createResizedImage (/Users/caigehui/Documents/ezz-example/app/extend/helper.js:106:16)
    at File.createFile (/Users/caigehui/Documents/ezz-example/app/service/base/file.js:31:30)
    at createFile.next (<anonymous>)
    at onFulfilled (/Users/caigehui/Documents/ezz-example/node_modules/co/index.js:65:19)
    at <anonymous>

pid: 9420
hostname: Mac-mini.local

2017-10-09 16:05:03,564 ERROR 9945 [-/127.0.0.1/-/49ms GET /api/files/59db027d9f334018d47acc38-small] nodejs.TypeError: Assignment to constant variable.
    at FilesController.show (/Users/caigehui/Documents/ezz-example/app/controller/base/files.js:23:13)
    at show.next (<anonymous>)
    at onFulfilled (/Users/caigehui/Documents/ezz-example/node_modules/co/index.js:65:19)
    at <anonymous>
    at process._tickCallback (internal/process/next_tick.js:188:7)

pid: 9945
hostname: Mac-mini.local

2017-10-09 16:35:06,988 ERROR 10698 [-/127.0.0.1/-/18ms GET /api/files/59db31e3a7cd1b28997998c7-small] nodejs.TypeError: Assignment to constant variable.
    at FilesController.show (/Users/caigehui/Documents/ezz-example/app/controller/base/files.js:32:28)
    at show.next (<anonymous>)
    at onFulfilled (/Users/caigehui/Documents/ezz-example/node_modules/co/index.js:65:19)
    at <anonymous>

pid: 10698
hostname: Mac-mini.local

2017-10-09 16:35:10,645 ERROR 10698 [-/127.0.0.1/-/3ms GET /api/files/59db31e3a7cd1b28997998c7-small] nodejs.TypeError: Assignment to constant variable.
    at FilesController.show (/Users/caigehui/Documents/ezz-example/app/controller/base/files.js:32:28)
    at show.next (<anonymous>)
    at onFulfilled (/Users/caigehui/Documents/ezz-example/node_modules/co/index.js:65:19)
    at <anonymous>

pid: 10698
hostname: Mac-mini.local

2017-10-09 16:35:12,352 ERROR 10698 [-/127.0.0.1/-/3ms GET /api/files/59db31e3a7cd1b28997998c7-small] nodejs.TypeError: Assignment to constant variable.
    at FilesController.show (/Users/caigehui/Documents/ezz-example/app/controller/base/files.js:32:28)
    at show.next (<anonymous>)
    at onFulfilled (/Users/caigehui/Documents/ezz-example/node_modules/co/index.js:65:19)
    at <anonymous>

pid: 10698
hostname: Mac-mini.local

2017-10-09 17:04:46,140 ERROR 11374 [-/127.0.0.1/-/11ms GET /api/files/59db31e3a7cd1b28997998c7-small] nodejs.TypeError: Cannot read property 'getFile' of undefined
    at FilesController.show (/Users/caigehui/Documents/ezz-example/app/controller/base/files.js:10:30)
    at show.next (<anonymous>)
    at onFulfilled (/Users/caigehui/Documents/ezz-example/node_modules/co/index.js:65:19)
    at /Users/caigehui/Documents/ezz-example/node_modules/co/index.js:54:5
    at Promise (<anonymous>)
    at Object.co (/Users/caigehui/Documents/ezz-example/node_modules/co/index.js:50:10)
    at Object.toPromise (/Users/caigehui/Documents/ezz-example/node_modules/co/index.js:118:63)
    at next (/Users/caigehui/Documents/ezz-example/node_modules/co/index.js:99:29)
    at onFulfilled (/Users/caigehui/Documents/ezz-example/node_modules/co/index.js:69:7)
    at /Users/caigehui/Documents/ezz-example/node_modules/co/index.js:54:5
    at Promise (<anonymous>)
    at Object.co (/Users/caigehui/Documents/ezz-example/node_modules/co/index.js:50:10)
    at Object.toPromise (/Users/caigehui/Documents/ezz-example/node_modules/co/index.js:118:63)
    at next (/Users/caigehui/Documents/ezz-example/node_modules/co/index.js:99:29)
    at onFulfilled (/Users/caigehui/Documents/ezz-example/node_modules/co/index.js:69:7)
    at /Users/caigehui/Documents/ezz-example/node_modules/co/index.js:54:5

pid: 11374
hostname: Mac-mini.local

2017-10-09 17:04:58,813 ERROR 11394 [-/127.0.0.1/-/34ms GET /api/files/59db31e3a7cd1b28997998c7-small] nodejs.ReferenceError: fs is not defined
    at File.getFile (/Users/caigehui/Documents/ezz-example/app/service/base/file.js:40:25)
    at getFile.next (<anonymous>)
    at onFulfilled (/Users/caigehui/Documents/ezz-example/node_modules/co/index.js:65:19)
    at <anonymous>

pid: 11394
hostname: Mac-mini.local

2017-10-09 17:04:58,956 ERROR 11394 [-/127.0.0.1/-/6ms GET /api/files/59db31e3a7cd1b28997998c7-small] nodejs.ReferenceError: fs is not defined
    at File.getFile (/Users/caigehui/Documents/ezz-example/app/service/base/file.js:40:25)
    at getFile.next (<anonymous>)
    at onFulfilled (/Users/caigehui/Documents/ezz-example/node_modules/co/index.js:65:19)
    at <anonymous>

pid: 11394
hostname: Mac-mini.local

2017-10-09 17:05:01,099 ERROR 11394 [-/127.0.0.1/-/3ms GET /api/files/59db31e3a7cd1b28997998c7-small] nodejs.ReferenceError: fs is not defined
    at File.getFile (/Users/caigehui/Documents/ezz-example/app/service/base/file.js:40:25)
    at getFile.next (<anonymous>)
    at onFulfilled (/Users/caigehui/Documents/ezz-example/node_modules/co/index.js:65:19)
    at <anonymous>

pid: 11394
hostname: Mac-mini.local

2017-10-09 17:05:06,157 ERROR 11394 [-/127.0.0.1/-/4ms GET /api/files/59db31e3a7cd1b28997998c7-small] nodejs.ReferenceError: fs is not defined
    at File.getFile (/Users/caigehui/Documents/ezz-example/app/service/base/file.js:40:25)
    at getFile.next (<anonymous>)
    at onFulfilled (/Users/caigehui/Documents/ezz-example/node_modules/co/index.js:65:19)
    at <anonymous>

pid: 11394
hostname: Mac-mini.local

2017-10-09 18:14:08,865 ERROR 13113 [-/127.0.0.1/-/33ms POST /api/files/verify] nodejs.TypeError: this.ctx.service.base.verifyFile is not a function
    at FilesController.verify (/Users/caigehui/Documents/ezz-example/app/controller/base/files.js:70:55)
    at verify.next (<anonymous>)
    at onFulfilled (/Users/caigehui/Documents/ezz-example/node_modules/co/index.js:65:19)
    at /Users/caigehui/Documents/ezz-example/node_modules/co/index.js:54:5
    at Promise (<anonymous>)
    at Object.co (/Users/caigehui/Documents/ezz-example/node_modules/co/index.js:50:10)
    at Object.toPromise (/Users/caigehui/Documents/ezz-example/node_modules/co/index.js:118:63)
    at next (/Users/caigehui/Documents/ezz-example/node_modules/co/index.js:99:29)
    at onFulfilled (/Users/caigehui/Documents/ezz-example/node_modules/co/index.js:69:7)
    at /Users/caigehui/Documents/ezz-example/node_modules/co/index.js:54:5
    at Promise (<anonymous>)
    at Object.co (/Users/caigehui/Documents/ezz-example/node_modules/co/index.js:50:10)
    at Object.toPromise (/Users/caigehui/Documents/ezz-example/node_modules/co/index.js:118:63)
    at next (/Users/caigehui/Documents/ezz-example/node_modules/co/index.js:99:29)
    at onFulfilled (/Users/caigehui/Documents/ezz-example/node_modules/co/index.js:69:7)
    at /Users/caigehui/Documents/ezz-example/node_modules/co/index.js:54:5

pid: 13113
hostname: Mac-mini.local

2017-10-09 18:14:42,256 ERROR 13146 [-/127.0.0.1/-/44ms POST /api/files/verify] nodejs.TypeError: Cannot read property '_id' of null
    at File.verifyFile (/Users/caigehui/Documents/ezz-example/app/service/base/file.js:70:25)
    at verifyFile.next (<anonymous>)
    at onFulfilled (/Users/caigehui/Documents/ezz-example/node_modules/co/index.js:65:19)
    at <anonymous>
    at process._tickCallback (internal/process/next_tick.js:188:7)

pid: 13146
hostname: Mac-mini.local

2017-10-09 18:52:22,014 ERROR 14359 [-/127.0.0.1/-/40ms GET /api/files/59db31e3a7cd1b28997998c7-small] nodejs.ReferenceError: fs is not defined
    at FilesController.show (/Users/caigehui/Documents/ezz-example/app/controller/base/files.js:39:25)
    at show.next (<anonymous>)
    at onFulfilled (/Users/caigehui/Documents/ezz-example/node_modules/co/index.js:65:19)
    at <anonymous>

pid: 14359
hostname: Mac-mini.local

